<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Differential gene analysis | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Differential gene analysis | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Differential gene analysis | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Yang Liu, Jennifer Altreuter, Yang Lin" />


<meta name="date" content="2022-12-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Preprocessing.html"/>
<link rel="next" href="Repertoire.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html"><i class="fa fa-check"></i><b>2</b> How to run RIMA</a><ul>
<li class="chapter" data-level="2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#install-rima-and-set-up-the-running-environment"><i class="fa fa-check"></i><b>2.1</b> Install RIMA and Set Up the Running Environment</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#download-pre-built-references"><i class="fa fa-check"></i><b>2.2</b> Download pre-built references</a><ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial"><i class="fa fa-check"></i><b>2.2.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#prepare-input-files"><i class="fa fa-check"></i><b>2.3</b> Prepare input files</a><ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial-1"><i class="fa fa-check"></i><b>2.3.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#running-rima"><i class="fa fa-check"></i><b>2.4</b> Running RIMA</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#output"><i class="fa fa-check"></i><b>2.5</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-alignment"><i class="fa fa-check"></i><b>3.1</b> Read Alignment</a></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a><ul>
<li class="chapter" data-level="3.4.1" data-path="Preprocessing.html"><a href="Preprocessing.html#start-with-tpm-matrix"><i class="fa fa-check"></i><b>3.4.1</b> Start with TPM matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-without-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 without batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.2</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.3</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.4" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.4</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> Cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-clonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Clonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#starting-with-tpm-matrix"><i class="fa fa-check"></i><b>6.7</b> Starting with TPM matrix</a><ul>
<li class="chapter" data-level="6.7.1" data-path="Infiltration.html"><a href="Infiltration.html#correlation-plots"><i class="fa fa-check"></i><b>6.7.1</b> Correlation Plots</a></li>
<li class="chapter" data-level="6.7.2" data-path="Infiltration.html"><a href="Infiltration.html#heatmaps-for-comparisons-across-cell-types"><i class="fa fa-check"></i><b>6.7.2</b> Heatmaps for comparisons across cell types</a></li>
<li class="chapter" data-level="6.7.3" data-path="Infiltration.html"><a href="Infiltration.html#box-plots-for-comparison-across-samples"><i class="fa fa-check"></i><b>6.7.3</b> Box plots for comparison across samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#characterizing-the-tme-and-t-cell-functionality"><i class="fa fa-check"></i><b>7.1</b> Characterizing the TME and T cell functionality</a></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.2</b> TIDE</a><ul>
<li class="chapter" data-level="7.2.1" data-path="Response.html"><a href="Response.html#config.yaml-sections-relevant-to-tide"><i class="fa fa-check"></i><b>7.2.1</b> Config.yaml sections relevant to TIDE</a></li>
<li class="chapter" data-level="7.2.2" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.2.2</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.2.3" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.2.3</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.2.4" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.2.4</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.2.5" data-path="Response.html"><a href="Response.html#normalization"><i class="fa fa-check"></i><b>7.2.5</b> Normalization</a></li>
<li class="chapter" data-level="7.2.6" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.2.6</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.3</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.4</b> Response comparison analysis of biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Typing &amp; Neoantigens</a><ul>
<li class="chapter" data-level="8.1" data-path="HLA.html"><a href="HLA.html#identify-hla-type"><i class="fa fa-check"></i><b>8.1</b> Identify HLA type</a></li>
<li class="chapter" data-level="8.2" data-path="HLA.html"><a href="HLA.html#hla-oncoplot"><i class="fa fa-check"></i><b>8.2</b> HLA Oncoplot</a></li>
<li class="chapter" data-level="8.3" data-path="HLA.html"><a href="HLA.html#neoantigen-identification"><i class="fa fa-check"></i><b>8.3</b> Neoantigen Identification</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>9</b> Fusion</a><ul>
<li class="chapter" data-level="9.1" data-path="fusion.html"><a href="fusion.html#fusion-calling"><i class="fa fa-check"></i><b>9.1</b> Fusion calling</a></li>
<li class="chapter" data-level="9.2" data-path="fusion.html"><a href="fusion.html#remove-fusion-homology"><i class="fa fa-check"></i><b>9.2</b> Remove fusion homology</a></li>
<li class="chapter" data-level="9.3" data-path="fusion.html"><a href="fusion.html#comparison-within-phenotype"><i class="fa fa-check"></i><b>9.3</b> Comparison within phenotype</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>10</b> Microbiome</a><ul>
<li class="chapter" data-level="10.1" data-path="microbiome.html"><a href="microbiome.html#centrifuge"><i class="fa fa-check"></i><b>10.1</b> Centrifuge</a></li>
<li class="chapter" data-level="10.2" data-path="microbiome.html"><a href="microbiome.html#microbial-classfication-from-rna-seq-data"><i class="fa fa-check"></i><b>10.2</b> Microbial classfication from RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html"><i class="fa fa-check"></i><b>11</b> Customize your own reference</a><ul>
<li class="chapter" data-level="11.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#reference-fasta"><i class="fa fa-check"></i><b>11.1</b> Reference fasta</a></li>
<li class="chapter" data-level="11.2" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gene-annotation-file-gtf"><i class="fa fa-check"></i><b>11.2</b> Gene annotation file (gtf)</a></li>
<li class="chapter" data-level="11.3" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-star-index"><i class="fa fa-check"></i><b>11.3</b> build STAR index</a></li>
<li class="chapter" data-level="11.4" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#rseqc-reference-files"><i class="fa fa-check"></i><b>11.4</b> RSeQC reference files</a></li>
<li class="chapter" data-level="11.5" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-salmon-index"><i class="fa fa-check"></i><b>11.5</b> build salmon index</a><ul>
<li class="chapter" data-level="11.5.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gmt-file-for-gene-set-analysis"><i class="fa fa-check"></i><b>11.5.1</b> GMT file for gene set analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#star-fusion-genome-resource-lib"><i class="fa fa-check"></i><b>11.6</b> STAR-Fusion genome resource lib</a></li>
<li class="chapter" data-level="11.7" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#centrifuge-index"><i class="fa fa-check"></i><b>11.7</b> Centrifuge index</a></li>
<li class="chapter" data-level="11.8" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#trust4-reference-files"><i class="fa fa-check"></i><b>11.8</b> TRUST4 reference files</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Differential" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Differential gene analysis</h1>
<p>Once you obtain gene counts from the preprocessing modules of RIMA, you can compare gene expression levels for different sample phenotypes. For example, you may wish to know which genes are differentially expressed between patients who respond and do not respond to a particular treatment. In addition, you may wish to know if particular gene sets are enriched in expression between different phenotypes. These gene sets may be associated with a known biological pathway, process or molecular function.</p>
<p>The differential gene analysis module of RIMA allows you to perform these comparisons.</p>
<div id="differential-gene-expression-using-deseq2" class="section level2">
<h2><span class="header-section-number">4.1</span> Differential gene expression using DESeq2</h2>
<p><a href='https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8'>DESeq2</a> is a commonly used package that provides methods for comparison of gene expression levels. The package is described in:</p>
<p>Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8</p>
<p>Details about how to run DeSeq2 are also included in the DESeq2
<a href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">vignette</a>.</p>
<div id="inputs-of-deseq2" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Inputs of DESeq2</h3>
<p>DESeq2 accepts four kinds of inputs: 1) transcript abundance files; 2) a read count matrix; 3) htseq-count files or 4) a SummarizedExperiment object.</p>
<p>Tximport is a tool that can import gene quantification data generated by Salmon (quant.sf) and convert the data into a gene-level abundance format that DESeq2 can use. RIMA generated quant.sf files during the preprocessing module.</p>
<p>As a part of using tximport, Ensemble transcript IDs are converted into gene names. RIMA contains a tx2gene.csv file with transcript IDs and gene names as below:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="Differential.html#cb28-1"></a>ID,GENENAME</span>
<span id="cb28-2"><a href="Differential.html#cb28-2"></a>ENST00000000233<span class="fl">.8</span>,ARF5</span>
<span id="cb28-3"><a href="Differential.html#cb28-3"></a>ENST00000000412<span class="fl">.6</span>,M6PR</span>
<span id="cb28-4"><a href="Differential.html#cb28-4"></a>ENST00000000442<span class="fl">.9</span>,ESRRA</span>
<span id="cb28-5"><a href="Differential.html#cb28-5"></a>ENST00000001008<span class="fl">.5</span>,FKBP4</span>
<span id="cb28-6"><a href="Differential.html#cb28-6"></a>ENST00000001146<span class="fl">.5</span>,CYP26B1</span>
<span id="cb28-7"><a href="Differential.html#cb28-7"></a>ENST00000002125<span class="fl">.7</span>,NDUFAF7</span>
<span id="cb28-8"><a href="Differential.html#cb28-8"></a>ENST00000002165<span class="fl">.9</span>,FUCA2</span>
<span id="cb28-9"><a href="Differential.html#cb28-9"></a>ENST00000002501<span class="fl">.9</span>,DBNDD1</span>
<span id="cb28-10"><a href="Differential.html#cb28-10"></a>ENST00000002596<span class="fl">.5</span>,HS3ST1</span></code></pre></div>
<p>Tximport uses the quant.sf files generated by Salmon and the tx2gene.csv file to generate matrices as described in DESeq2:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="Differential.html#cb29-1"></a><span class="co">### filelist is a list of quant.sf files generated by Salmon gene quantification.</span></span>
<span id="cb29-2"><a href="Differential.html#cb29-2"></a>txi &lt;-<span class="st"> </span><span class="kw">tximport</span>(filelist, <span class="dt">type=</span>Salmon, <span class="dt">tx2gene=</span>tx2gene)</span></code></pre></div>
</div>
</div>
<div id="different-gene-expression-analysis" class="section level2">
<h2><span class="header-section-number">4.2</span> Different gene expression analysis</h2>
<p>After creating the transcript abundance file, we are ready to find differentially expressed genes between your selected conditions. If you have identified a batch effect in your earlier analysis, DESeq2 can be run to account for this batch effect. We will first describe how to run DESeq2 without a batch effect. We will then describe how to run DESeq2 when a batch effect exists.</p>
<div id="running-deseq2-without-batch-effect" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Running DESeq2 without batch effect</h3>
<p>We will run DESeq2 to determine differential expression for a particular phenotype. In our case, we are interested in the genes that are differentially expressed between responders and non-responders. For this example, the Group column in the metasheet.csv has been set up with “R”, “NR” or “NA” as shown in Chapter 2.3 and the config.yaml file is set up with</p>
<ul>
<li>the “batch” parameter set to false;</li>
<li>“design” set to Group</li>
<li>“Treatment” set to R</li>
<li>“Control” set to NR</li>
</ul>
<p>as seen in Chapter 2.3.</p>
<p>This results in a DESeq call like the following:</p>
<pre><code>ddsTxi &lt;- DESeqDataSetFromTximport(txi,
                                       colData = samples,
                                       design = ~ Group)

dds &lt;- DESeq(ddsTxi)</code></pre>
</div>
<div id="running-deseq2-with-batch-effect" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Running DESeq2 with batch effect</h3>
<p>DESeq2 can account for any batch effect by providing the batch variable as a covariate. If your batch effect analysis from the preprocessing module indicated that there is a batch effect in your samples, set the “batch” field in config.yaml to the appropriate column name in your metasheet.</p>
<pre><code>#########Cohort level analysis parameters################
design: Group             # Condition on which to do comparsion (as set up in metasheet.csv)
Treatment: R              # Treatment use in DESeq2, corresponding to positive log fold change
Control: NR               # Control use in DESeq2, corresponding to negative log fold change
batch: syn_batch          # Options: &#39;false&#39; or a column name from the metasheet.csv.  </code></pre>
<p>We use the information in “batch” for DESeq2 in the following manner, where “syn_batch” is the column designated in the config.yaml file:</p>
<pre><code>ddsTxi &lt;- DESeqDataSetFromTximport(txi,
                                       colData = samples,
                                       design = ~ syn_batch + Group)
dds &lt;- DESeq(ddsTxi)</code></pre>
</div>
<div id="output-from-deseq2" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Output from DESeq2</h3>
<p>DESeq2 output includes values for baseMean, log2FoldChange, lfcSE, stat, pvalue, padj. These fields have the following meanings:</p>
<ul>
<li>baseMean = the average of the normalized counts for all samples</li>
<li>Log2FoldChange = log base 2 fold changes for the condition tested. For example if the file is Responder_responder_VS_nonresponder_DESqe2_raw.txt, Log2FoldChange will be the estimated logarithmic fold change log2(responder/nonresponder).</li>
<li>lfcSE = Log2FoldChange Standard Error</li>
<li>stat= Wald statistic</li>
<li>pvalue = Wald test p value for the Log2FoldChange estimate</li>
<li>padj. = FDR adjusted p value (Benjamini-Hochberg adjusted p value)</li>
</ul>
<p>{comparison condition}_DEseq2_raw.txt lists Gene names directly from the DESeq2 output. It also contains a value for -log10(padj).</p>
</div>
<div id="volcano-plots" class="section level3">
<h3><span class="header-section-number">4.2.4</span> Volcano Plot(s)</h3>
<p>Volcano plots are often used to visualize the results of differential expression analysis. The x axis represents the log2FC, and the y axis represent the -log10(P.value). Up-regulated genes have positive log2FCs and down-regulated genes have negative log2FCs.<br />
</p>
<p>Below is an example of a volcano plot from the previous differential expression analysis. Genes that meet the following criteria are labeled:</p>
<ul>
<li>FDR &lt; 0.05</li>
<li>absolute Log2FoldChange &gt; 1</li>
</ul>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-16"></span>
<img src="images/volcano.png" alt="Volcano plot"  />
<p class="caption">
Figure 4.1: Volcano plot
</p>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level2">
<h2><span class="header-section-number">4.3</span> Gene set enrichment analysis (GSEA)</h2>
<p>GSEA was developed by UC San Diego and the Broad Institute. It is described in</p>
<p>Subramanian, Tamayo, et al. (2005, PNAS 102, 15545-15550) and Mootha, Lindgren, et al. (2003, Nat Genet 34, 267-273).</p>
<p>Gene Set Enrichment Analysis can be used to detect patterns in differential expression that affect particular gene pathways, molecular functions, cellular components or processes. RIMA uses the <a href='https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html'>clusterProfile</a> R package to look for enrichment in the following gene sets:</p>
<ul>
<li><a href="https://www.genome.jp/kegg/">KEGG canonical pathways</a></li>
<li><a href="http://geneontology.org/">Gene Ontology</a> – including Biological Process (BP), Cellular Component (CC) and Molecular Function (MF).</li>
<li><a href="https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp#H">MSigDB Hallmark gene sets</a></li>
</ul>
<p>RIMA uses a ranked list of DESeq2 outputs. Ranking is based on the following:</p>
<pre><code>sign(data$log2FoldChange) * (-log10(data$pvalue))</code></pre>
<p>GSEA output is reported in both text and figure outputs:</p>
<p><strong>Example output folder structure</strong>
<img src="images/Output_gsea.png" width="50%" style="display: block; margin: auto auto auto 0;" /></p>
<p>Each of these tab separated files contains the following information:</p>
<ul>
<li>ID = ID of the gene set</li>
<li>Description = description of the gene set</li>
<li>setSize = number of genes in the gene set after filtering out genes not in the expression data.</li>
<li>enrichmentScore = the degree to which the gene set is enriched in the top or bottom of the ranked set of genes.</li>
<li>NES = normalized gene enrichment score.<br />
</li>
<li>pvalue = nominal p value</li>
<li>p.adjust = adjusted p value; RIMA sets pAdjustMethod to ‘BH’ (Bejamini Hochberg) when running clusterProfiler</li>
<li>qvalues = q value (FDR as implemented by GSEA)</li>
<li>rank = the position in the ranked list at which the maximal enrichment score was achieved.</li>
<li>leading_edge = 3 statistics (tags, list and signal) on the subset of genes that most influence the enrichment score</li>
<li>core_enrichment group = the subset of genes that contribute most to the enrichment score.</li>
<li>significance = a representation of the p.adjust value using stars. One star is set for p.adjust &lt;0.1, two stars &lt;0.5 and three stars &lt; 0.01.</li>
</ul>
<p>Please see the <a href="https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/">clusterProfile tutorial</a> for more information about these values.</p>
<p>Below is an example of the figure output for GSEA using the Hallmark genesets.<br />
</p>
<strong>Example GSEA figure output</strong>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-18"></span>
<img src="images/Group_R_vs_NR_HALLMARK_terms.png" alt="Hallmark gene set enrichment output" width="120%" />
<p class="caption">
Figure 4.2: Hallmark gene set enrichment output
</p>
</div>
</div>
<div id="single-sample-gene-set-enrichment-analysis-ssgsea" class="section level2">
<h2><span class="header-section-number">4.4</span> Single sample gene set enrichment analysis (ssGSEA)</h2>
<p>ssGSEA is a different method of trying to find enriched pathways in your dataset. GSEA uses the differential expression patterns of all samples to generate enrichment scores based on phenotypic differences. In contrast, ssGSEA generates enrichment scores for each sample. The ssGSEA results can then be used to look for outliers in your dataset or to compare enriched pathways among samples. Please see <a href="https://gsea-msigdb.github.io/ssGSEA-gpmodule/v10/index.html">ssGSEA</a> for more information.</p>
<p>The inputs to ssGSEA are log2 transformed tpms with batch effects removed if needed. RIMA provides ssgsea results for the KEGG and Hallmark gene sets.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Preprocessing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Repertoire.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
