<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Immune Infiltration | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Yang Liu, Jennifer Altreuter, Yang Lin" />


<meta name="date" content="2022-12-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Repertoire.html"/>
<link rel="next" href="Response.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html"><i class="fa fa-check"></i><b>2</b> How to run RIMA</a><ul>
<li class="chapter" data-level="2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#install-rima-and-set-up-the-running-environment"><i class="fa fa-check"></i><b>2.1</b> Install RIMA and Set Up the Running Environment</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#download-pre-built-references"><i class="fa fa-check"></i><b>2.2</b> Download pre-built references</a><ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial"><i class="fa fa-check"></i><b>2.2.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#prepare-input-files"><i class="fa fa-check"></i><b>2.3</b> Prepare input files</a><ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial-1"><i class="fa fa-check"></i><b>2.3.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#running-rima"><i class="fa fa-check"></i><b>2.4</b> Running RIMA</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#output"><i class="fa fa-check"></i><b>2.5</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-alignment"><i class="fa fa-check"></i><b>3.1</b> Read Alignment</a></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a><ul>
<li class="chapter" data-level="3.4.1" data-path="Preprocessing.html"><a href="Preprocessing.html#start-with-tpm-matrix"><i class="fa fa-check"></i><b>3.4.1</b> Start with TPM matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-without-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 without batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.2</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.3</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.4" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.4</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> Cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-clonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Clonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#starting-with-tpm-matrix"><i class="fa fa-check"></i><b>6.7</b> Starting with TPM matrix</a><ul>
<li class="chapter" data-level="6.7.1" data-path="Infiltration.html"><a href="Infiltration.html#correlation-plots"><i class="fa fa-check"></i><b>6.7.1</b> Correlation Plots</a></li>
<li class="chapter" data-level="6.7.2" data-path="Infiltration.html"><a href="Infiltration.html#heatmaps-for-comparisons-across-cell-types"><i class="fa fa-check"></i><b>6.7.2</b> Heatmaps for comparisons across cell types</a></li>
<li class="chapter" data-level="6.7.3" data-path="Infiltration.html"><a href="Infiltration.html#box-plots-for-comparison-across-samples"><i class="fa fa-check"></i><b>6.7.3</b> Box plots for comparison across samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#characterizing-the-tme-and-t-cell-functionality"><i class="fa fa-check"></i><b>7.1</b> Characterizing the TME and T cell functionality</a></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.2</b> TIDE</a><ul>
<li class="chapter" data-level="7.2.1" data-path="Response.html"><a href="Response.html#config.yaml-sections-relevant-to-tide"><i class="fa fa-check"></i><b>7.2.1</b> Config.yaml sections relevant to TIDE</a></li>
<li class="chapter" data-level="7.2.2" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.2.2</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.2.3" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.2.3</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.2.4" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.2.4</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.2.5" data-path="Response.html"><a href="Response.html#normalization"><i class="fa fa-check"></i><b>7.2.5</b> Normalization</a></li>
<li class="chapter" data-level="7.2.6" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.2.6</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.3</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.4</b> Response comparison analysis of biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Typing &amp; Neoantigens</a><ul>
<li class="chapter" data-level="8.1" data-path="HLA.html"><a href="HLA.html#identify-hla-type"><i class="fa fa-check"></i><b>8.1</b> Identify HLA type</a></li>
<li class="chapter" data-level="8.2" data-path="HLA.html"><a href="HLA.html#hla-oncoplot"><i class="fa fa-check"></i><b>8.2</b> HLA Oncoplot</a></li>
<li class="chapter" data-level="8.3" data-path="HLA.html"><a href="HLA.html#neoantigen-identification"><i class="fa fa-check"></i><b>8.3</b> Neoantigen Identification</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>9</b> Fusion</a><ul>
<li class="chapter" data-level="9.1" data-path="fusion.html"><a href="fusion.html#fusion-calling"><i class="fa fa-check"></i><b>9.1</b> Fusion calling</a></li>
<li class="chapter" data-level="9.2" data-path="fusion.html"><a href="fusion.html#remove-fusion-homology"><i class="fa fa-check"></i><b>9.2</b> Remove fusion homology</a></li>
<li class="chapter" data-level="9.3" data-path="fusion.html"><a href="fusion.html#comparison-within-phenotype"><i class="fa fa-check"></i><b>9.3</b> Comparison within phenotype</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>10</b> Microbiome</a><ul>
<li class="chapter" data-level="10.1" data-path="microbiome.html"><a href="microbiome.html#centrifuge"><i class="fa fa-check"></i><b>10.1</b> Centrifuge</a></li>
<li class="chapter" data-level="10.2" data-path="microbiome.html"><a href="microbiome.html#microbial-classfication-from-rna-seq-data"><i class="fa fa-check"></i><b>10.2</b> Microbial classfication from RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html"><i class="fa fa-check"></i><b>11</b> Customize your own reference</a><ul>
<li class="chapter" data-level="11.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#reference-fasta"><i class="fa fa-check"></i><b>11.1</b> Reference fasta</a></li>
<li class="chapter" data-level="11.2" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gene-annotation-file-gtf"><i class="fa fa-check"></i><b>11.2</b> Gene annotation file (gtf)</a></li>
<li class="chapter" data-level="11.3" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-star-index"><i class="fa fa-check"></i><b>11.3</b> build STAR index</a></li>
<li class="chapter" data-level="11.4" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#rseqc-reference-files"><i class="fa fa-check"></i><b>11.4</b> RSeQC reference files</a></li>
<li class="chapter" data-level="11.5" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-salmon-index"><i class="fa fa-check"></i><b>11.5</b> build salmon index</a><ul>
<li class="chapter" data-level="11.5.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gmt-file-for-gene-set-analysis"><i class="fa fa-check"></i><b>11.5.1</b> GMT file for gene set analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#star-fusion-genome-resource-lib"><i class="fa fa-check"></i><b>11.6</b> STAR-Fusion genome resource lib</a></li>
<li class="chapter" data-level="11.7" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#centrifuge-index"><i class="fa fa-check"></i><b>11.7</b> Centrifuge index</a></li>
<li class="chapter" data-level="11.8" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#trust4-reference-files"><i class="fa fa-check"></i><b>11.8</b> TRUST4 reference files</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Infiltration" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Immune Infiltration</h1>
<p><strong>Cell infiltration estimation in the tumor microenvironment</strong><br />
The tumor microenviroment (<strong>TME</strong>) is a complicated system consisting of immune cells, stromal cells, and extracellular factors. Estimating cell components is essential for classifying the distinct tumor immune microenvironment (TIME) phenotype. Furthermore, dissecting the TIME by evaluating cell components plays a significant role in untangling the mechanisms of tumor progression and immune evasion.</p>
<p>Traditionally, there have been two major approaches for estimating cell infiltration in the TME – 1) deconvolution-based and 2) marker-based approaches. Deconvolution-based methods, such as TIMER, quanTIseq, EPIC, and CIBERSORT, consider a given gene expression profile as a linear combination of pre-defined immune gene signatures present at different ratios. A linear regression model is often applied to estimate the coefficients of genes, which are later used to infer immune cell abundances or fractions. Marker-based approaches, such as xCell and MCP-counter, quantify the signature enrichment score of a list of cell-specific marker genes from gene expression profiles. <a href="https://github.com/icbi-lab/immunedeconv">ImmuneDeconv</a> is an R package that implements these six algorithms. All six algorithms estimate cell infiltration of immune cells, while three of the algorithms also estimate stromal cells. EPIC and quanTIseq also assess uncharacterized cells defined as cancer cells. Of note, CIBERSORT absolute mode, EPIC, and quanTIseq support inter-comparison between sample groups and intra-comparison between cell types. TIMER, xCell, and MCP-counter only support inter-comparison between sample groups within the same cell type. Immune cell deconvolution is still an open question in the field, and different algorithms could give slightly different results. A user may wish to evaluate the consistency of predictions stemming from different algorithms, the agreement of results with estimations from other modalities, and/or the derivation of the marker genes used for the tool. Newer algorithms are being developed which can utilize single-cell RNA-seq data as reference and which attempt to impute cell type-specific gene expression. These tools may improve predictions, especially if a well-annotated and compatible reference is available.<br />
</p>
<div id="cibersort" class="section level2">
<h2><span class="header-section-number">6.1</span> Cibersort</h2>
<p><em>Note: RIMA uses Cibersort in the immune_infiltration modules. Use of Cibersort requires registration for use of their gene signature matrix. In order to run the immune_infiltration module of RIMA, you must register and download the LM22.txt matrix from the <a href="https://cibersort.stanford.edu/">CIBERSORT website </a>. Once you obtain this file, add it to the ~/RIMA/RIMA_pipeline/static/cibersort folder.</em></p>
<p><a href="https://www.nature.com/articles/nmeth.3337">CIBERSORT</a> is a deconvolution-based method that uses the v-support vector regression method (v-SVR) to estimate each immune cell type’s relative proportion from a gene expression profile (GEP). It requires an input of reference gene expression signatures. CIBERSORT provides a gene signature matrix (LM22.txt) which contains 547 genes and distinguishes 7 T-cell types, naïve and memory B cells, plasma cells, and NK cells. The CIBERSORT absolute mode scales cellular fractions to a score that reflects each cell type’s absolute proportion, allowing comparison across both samples and cell types.
The <strong>CIBERSORT source code</strong> and <strong>LM22 signature matrix</strong> are only freely available to academic users – register for access on the <a href="https://cibersort.stanford.edu/">CIBERSORT website </a>.<br />
</p>
<p>To run CIBERSORT using ImmuneDeconv:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="Infiltration.html#cb60-1"></a><span class="co">#demo</span></span>
<span id="cb60-2"><a href="Infiltration.html#cb60-2"></a><span class="kw">suppressMessages</span>(<span class="kw">library</span>(immunedeconv))</span>
<span id="cb60-3"><a href="Infiltration.html#cb60-3"></a><span class="co">#read gene expression matrix </span></span>
<span id="cb60-4"><a href="Infiltration.html#cb60-4"></a>input &lt;-<span class="st"> &quot;tpm_convertID.txt&quot;</span></span>
<span id="cb60-5"><a href="Infiltration.html#cb60-5"></a></span>
<span id="cb60-6"><a href="Infiltration.html#cb60-6"></a><span class="co">#load CIBERSORT source code and signature matrix </span></span>
<span id="cb60-7"><a href="Infiltration.html#cb60-7"></a><span class="kw">source</span>(<span class="st">&quot;CIBERSORT.R&quot;</span>)</span>
<span id="cb60-8"><a href="Infiltration.html#cb60-8"></a>sig_matrix &lt;-<span class="st"> &quot;LM22.txt&quot;</span></span>
<span id="cb60-9"><a href="Infiltration.html#cb60-9"></a></span>
<span id="cb60-10"><a href="Infiltration.html#cb60-10"></a><span class="co">#Run CIBERSORT abs </span></span>
<span id="cb60-11"><a href="Infiltration.html#cb60-11"></a><span class="co">#The number of permutation</span></span>
<span id="cb60-12"><a href="Infiltration.html#cb60-12"></a>cibersort_perm =<span class="st"> </span><span class="dv">100</span></span>
<span id="cb60-13"><a href="Infiltration.html#cb60-13"></a><span class="co">#Quantile normalization of input mixture, default = FALSE for RNA-Seq data</span></span>
<span id="cb60-14"><a href="Infiltration.html#cb60-14"></a>cibersort_qn =<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb60-15"><a href="Infiltration.html#cb60-15"></a><span class="co">#whether to apply absolute mode in cibersort</span></span>
<span id="cb60-16"><a href="Infiltration.html#cb60-16"></a>cibersort_abs =<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb60-17"><a href="Infiltration.html#cb60-17"></a><span class="co">#sig.score = for each mixture sample, define S as the median expression,level of all genes in the signature matrix divided by the median expression level of all genes in the mixture. Multiple cell subset fractions by S.</span></span>
<span id="cb60-18"><a href="Infiltration.html#cb60-18"></a>cibersort_abs_method =<span class="st"> &quot;sig.score&quot;</span></span>
<span id="cb60-19"><a href="Infiltration.html#cb60-19"></a>res_ciber &lt;-<span class="st"> </span><span class="kw">CIBERSORT</span>(sig_matrix, input, <span class="dt">perm =</span> cibersort_perm, <span class="dt">QN =</span> cibersort_qn, <span class="dt">absolute =</span> cibersort_abs,</span>
<span id="cb60-20"><a href="Infiltration.html#cb60-20"></a>                       <span class="dt">abs_method =</span> cibersort_abs_method)</span>
<span id="cb60-21"><a href="Infiltration.html#cb60-21"></a></span>
<span id="cb60-22"><a href="Infiltration.html#cb60-22"></a><span class="kw">head</span>(res_ciber,<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##            B cells naive B cells memory Plasma cells T cells CD8
## SRR8281218   0.002023271    0.003060637  0.000000000  0.12297118
## SRR8281219   0.010693087    0.000000000  0.001201108  0.38499197
## SRR8281226   0.000000000    0.008281786  0.000000000  0.01442307
##            T cells CD4 naive T cells CD4 memory resting
## SRR8281218                 0                 0.14975225
## SRR8281219                 0                 0.22216858
## SRR8281226                 0                 0.08219228
##            T cells CD4 memory activated T cells follicular helper
## SRR8281218                            0                0.08924869
## SRR8281219                            0                0.16924345
## SRR8281226                            0                0.06671124
##            T cells regulatory (Tregs) T cells gamma delta NK cells resting
## SRR8281218                 0.09617560                   0       0.06763887
## SRR8281219                 0.00000000                   0       0.00000000
## SRR8281226                 0.04296535                   0       0.07211706
##            NK cells activated  Monocytes Macrophages M0 Macrophages M1
## SRR8281218          0.0000000 0.06195550      0.0000000     0.01550602
## SRR8281219          0.1076938 0.03659288      0.1552708     0.03207925
## SRR8281226          0.0357515 0.41486727      0.0000000     0.00000000
##            Macrophages M2 Dendritic cells resting Dendritic cells activated
## SRR8281218      0.7049077                       0               0.000000000
## SRR8281219      1.1589577                       0               0.000000000
## SRR8281226      0.6474766                       0               0.004669831
##            Mast cells resting Mast cells activated Eosinophils Neutrophils
## SRR8281218         0.11994310                    0 0.003766691  0.00000000
## SRR8281219         0.24218289                    0 0.000000000  0.07942792
## SRR8281226         0.01289667                    0 0.002926060  0.12361329
##            P-value Correlation     RMSE Absolute score (sig.score)
## SRR8281218    0.07  0.15699314 1.027475                   1.436950
## SRR8281219    0.24  0.05195861 1.072383                   2.600503
## SRR8281226    0.08  0.13503937 1.054532                   1.528892</code></pre>
</div>
<div id="timer" class="section level2">
<h2><span class="header-section-number">6.2</span> TIMER</h2>
<p><a href="http://cistrome.org/TIMER/">TIMER</a> (Tumor IMmune Estimation Resource) uses linear least square regression to estimate six tumor-infiltrating immune cell types (B, CD4T, CD8T, neutrophils, macrophages, and dendritic cells). Since the tumor purity is very different across different cancer cell types, TIMER selects specific genes as immune signatures based on the correlation between gene expression and tumor purity. (see <a href="https://pubmed.ncbi.nlm.nih.gov/27549193/">Bo Li, et al.</a> for more information.<br />
</p>
<p>Tutorial of TIMER’s web interface:<br />
<iframe width="560" height="315" src="https://www.youtube.com/embed/94v8XboCrXU" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe><br />
</p>
<p>Tutorial of TIMER2:<br />
</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/2hmxioq1pJo" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><br />
</p>
<p>Note that the TIMER score can only be compared within samples, and the current version of TIMER supports 23 different TCGA cancer types.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="Infiltration.html#cb62-1"></a><span class="co">#load gene expression matrix </span></span>
<span id="cb62-2"><a href="Infiltration.html#cb62-2"></a>df &lt;-<span class="st"> </span><span class="kw">read.table</span>(input, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="dt">row.names =</span> <span class="dv">1</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)</span>
<span id="cb62-3"><a href="Infiltration.html#cb62-3"></a></span>
<span id="cb62-4"><a href="Infiltration.html#cb62-4"></a><span class="co">#The TCGA cancer acronyms of demo data</span></span>
<span id="cb62-5"><a href="Infiltration.html#cb62-5"></a>cancertype &lt;-<span class="st"> &quot;GBM&quot;</span></span>
<span id="cb62-6"><a href="Infiltration.html#cb62-6"></a><span class="co">#Available cancer type </span></span>
<span id="cb62-7"><a href="Infiltration.html#cb62-7"></a>immunedeconv<span class="op">::</span>timer_available_cancers</span>
<span id="cb62-8"><a href="Infiltration.html#cb62-8"></a>res_timer =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;timer&quot;</span>,<span class="dt">indications=</span><span class="kw">rep</span>(<span class="kw">tolower</span>(cancertype),<span class="kw">ncol</span>(df))))</span></code></pre></div>
</div>
<div id="quantiseq" class="section level2">
<h2><span class="header-section-number">6.3</span> quanTIseq</h2>
<p><a href="https://icbi.i-med.ac.at/software/quantiseq/doc/index.html">quanTIseq</a> uses constrained least square regression to estimate absolute proportions of the infiltration levels of 10 immune cell types from RNA seq data. Therefore, the score of quanTIseq allows inter- or intra-sample comparison. quanTIseq provides an entire pipeline that can either process the pre-computed gene expression matrix (TPM matrix) or directly analyze bulk RNA-Seq data. RIMA uses the TPM matrix generated in the proprocessing modules for all immune infiltration algorithms.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="Infiltration.html#cb63-1"></a><span class="co">#Run quanTIseq </span></span>
<span id="cb63-2"><a href="Infiltration.html#cb63-2"></a>res_quant =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;quantiseq&quot;</span>))</span></code></pre></div>
</div>
<div id="xcell" class="section level2">
<h2><span class="header-section-number">6.4</span> xCell</h2>
<p><a href="https://xcell.ucsf.edu">xCell</a> performs single-sample gene set enrichment analysis (ssGSEA) on expression data to evaluate the enrichment of immune marker genes. These marker genes were validated using cytometry immunophenotyping and in-silico simulations. Note that results of xCell depend on the number of samples. The power of estimation might be lower if your dataset has limited non-heterogeneous samples.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="Infiltration.html#cb64-1"></a><span class="co">#Run xCell</span></span>
<span id="cb64-2"><a href="Infiltration.html#cb64-2"></a>res_xcell =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;xcell&quot;</span>))</span></code></pre></div>
</div>
<div id="epic" class="section level2">
<h2><span class="header-section-number">6.5</span> EPIC</h2>
<p><a href="https://gfellerlab.shinyapps.io/EPIC_1-1/">EPIC</a> uses constrained least square regression to estimate six immune cell types, fibroblasts, and endothelial cells. EPIC collects a unique gene expression reference from circulating and tumor-infiltrating cells. Further, it extended its algorithm to evaluate the uncharacterized cancer cells. The score that comes from the EPIC algorithm is an absolute value that can be compared within or across samples.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="Infiltration.html#cb65-1"></a><span class="co">#Run EPIC</span></span>
<span id="cb65-2"><a href="Infiltration.html#cb65-2"></a>res_epic =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;epic&quot;</span>))</span></code></pre></div>
</div>
<div id="mcp-counter" class="section level2">
<h2><span class="header-section-number">6.6</span> MCP-counter</h2>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1070-5">MCP-counter</a> uses its transcriptomic markers (TM) curated from 81 public datasets to estimate fibroblasts, endothelial cells, and eight immune cell types.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="Infiltration.html#cb66-1"></a><span class="co">#Run EPIC</span></span>
<span id="cb66-2"><a href="Infiltration.html#cb66-2"></a>res_mcp =<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">deconvolute</span>(df, <span class="st">&quot;mcp_counter&quot;</span>))</span></code></pre></div>
</div>
<div id="starting-with-tpm-matrix" class="section level2">
<h2><span class="header-section-number">6.7</span> Starting with TPM matrix</h2>
<p><a href="https://github.com/liulab-dfci/RIMA_pipeline/blob/master/src/immune_infiltration/immune_infiltration.R">immune_infiltration.R</a> contains R scripts for immune infiltration comparison analysis using a TPM matrix as input. Below is an example of running the scripts to obtain CIBERSORT absolute results:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="Infiltration.html#cb67-1"></a>Rscript src<span class="op">/</span>immune_infiltration<span class="op">/</span>immune_infiltration.R <span class="op">-</span>e exprsn <span class="op">-</span>t cancer_type <span class="op">-</span>p permutation <span class="op">-</span>q <span class="ot">FALSE</span> <span class="op">-</span>a <span class="ot">TRUE</span> <span class="op">-</span>m abs_method <span class="op">-</span>o out_dir</span>
<span id="cb67-2"><a href="Infiltration.html#cb67-2"></a></span>
<span id="cb67-3"><a href="Infiltration.html#cb67-3"></a>Usage<span class="op">:</span></span>
<span id="cb67-4"><a href="Infiltration.html#cb67-4"></a><span class="st">  </span><span class="op">-</span>e expression file [Required]</span>
<span id="cb67-5"><a href="Infiltration.html#cb67-5"></a>  <span class="op">-</span>o output directory [Required]</span>
<span id="cb67-6"><a href="Infiltration.html#cb67-6"></a>  <span class="op">-</span>p permutation <span class="cf">for</span> cibersort</span>
<span id="cb67-7"><a href="Infiltration.html#cb67-7"></a>  <span class="op">-</span>q whether enable quantile normalization <span class="cf">for</span> cibersort <span class="ot">TRUE</span><span class="op">/</span><span class="ot">FALSE</span></span>
<span id="cb67-8"><a href="Infiltration.html#cb67-8"></a>  <span class="op">-</span>a whether enable absolute mode of cibersort <span class="ot">TRUE</span><span class="op">/</span><span class="ot">FALSE</span></span>
<span id="cb67-9"><a href="Infiltration.html#cb67-9"></a>  <span class="op">-</span>m absolute method <span class="cf">for</span> cibersort_abs Default<span class="op">:</span><span class="st"> &#39;sig.score&#39;</span></span>
<span id="cb67-10"><a href="Infiltration.html#cb67-10"></a>  <span class="op">-</span>t cancer type <span class="cf">for</span> timer</span></code></pre></div>
<p>RIMA will generate immune infiltration results for all six algorithms which can be used to compare immune cell levels between phenotypes. After RIMA’s immune_infiltration module is run, text output files as well as boxplot, correlation and heatmap plots are stored in the analysis/immune_infiltration folder of the RIMA_pipeline. Example code is provided below for heatmap and boxplots like those generated by RIMA. CIBERSORT abs results are used as an example. If you wish to use this code to generate new figures, set “phenotype” in the hmap and boxfig calls to the appropriate column in metasheet.csv.</p>
<div id="correlation-plots" class="section level3">
<h3><span class="header-section-number">6.7.1</span> Correlation Plots</h3>
<p>RIMA generates correlation plots for 8 cell types. (B cells, CD4 T cells, CD8 Tcells, DCs, Macrophages, Treg cells, NK cells and Neutrophils) These image files display Spearman correlation values for all six algorithms, allowing you to compare cell estimations between algorithms.</p>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-35"></span>
<img src="images/Group_syn_batch_CD8_T_cell_corr.png" alt="Volcano plot"  />
<p class="caption">
Figure 6.1: Volcano plot
</p>
</div>
</div>
<div id="heatmaps-for-comparisons-across-cell-types" class="section level3">
<h3><span class="header-section-number">6.7.2</span> Heatmaps for comparisons across cell types</h3>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="Infiltration.html#cb68-1"></a><span class="co">#extract the immune cell score from results of CIBERSORT</span></span>
<span id="cb68-2"><a href="Infiltration.html#cb68-2"></a>res_ciber &lt;-<span class="st"> </span>res_ciber[,<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>]</span>
<span id="cb68-3"><a href="Infiltration.html#cb68-3"></a>res_ciber</span></code></pre></div>
<pre><code>##            B cells naive B cells memory Plasma cells T cells CD8
## SRR8281218   0.002023271    0.003060637  0.000000000  0.12297118
## SRR8281219   0.010693087    0.000000000  0.001201108  0.38499197
## SRR8281226   0.000000000    0.008281786  0.000000000  0.01442307
## SRR8281236   0.119197082    0.000000000  0.175028807  0.05841897
## SRR8281230   0.000000000    0.001725996  0.000000000  0.02099308
## SRR8281233   0.074569126    0.000000000  0.022548328  0.09612605
## SRR8281244   0.000000000    0.133329588  0.041170411  0.58644201
## SRR8281245   0.057960600    0.000000000  0.001729094  0.71691961
## SRR8281243   0.021015715    0.000000000  0.000000000  0.05560813
## SRR8281251   0.000000000    0.014652173  0.000000000  0.12265135
## SRR8281238   0.010961877    0.000000000  0.000000000  0.07328546
## SRR8281250   0.000000000    0.007211828  0.000000000  0.11124637
##            T cells CD4 naive T cells CD4 memory resting
## SRR8281218                 0               0.1497522454
## SRR8281219                 0               0.2221685808
## SRR8281226                 0               0.0821922827
## SRR8281236                 0               0.0616983261
## SRR8281230                 0               0.2636255094
## SRR8281233                 0               0.0540161771
## SRR8281244                 0               0.0000000000
## SRR8281245                 0               0.1346463483
## SRR8281243                 0               0.2983063326
## SRR8281251                 0               0.0008469304
## SRR8281238                 0               0.1462044340
## SRR8281250                 0               0.0912421429
##            T cells CD4 memory activated T cells follicular helper
## SRR8281218                            0                0.08924869
## SRR8281219                            0                0.16924345
## SRR8281226                            0                0.06671124
## SRR8281236                            0                0.02097714
## SRR8281230                            0                0.04434860
## SRR8281233                            0                0.01702242
## SRR8281244                            0                0.20198021
## SRR8281245                            0                0.08491094
## SRR8281243                            0                0.02547211
## SRR8281251                            0                0.13107578
## SRR8281238                            0                0.06549731
## SRR8281250                            0                0.08310786
##            T cells regulatory (Tregs) T cells gamma delta NK cells resting
## SRR8281218                0.096175601                   0       0.06763887
## SRR8281219                0.000000000                   0       0.00000000
## SRR8281226                0.042965351                   0       0.07211706
## SRR8281236                0.016105425                   0       0.01389437
## SRR8281230                0.003663422                   0       0.06303278
## SRR8281233                0.000000000                   0       0.00000000
## SRR8281244                0.000000000                   0       0.00000000
## SRR8281245                0.064756411                   0       0.00000000
## SRR8281243                0.024209477                   0       0.02796011
## SRR8281251                0.000000000                   0       0.03345416
## SRR8281238                0.000000000                   0       0.00000000
## SRR8281250                0.021319048                   0       0.11199972
##            NK cells activated    Monocytes Macrophages M0 Macrophages M1
## SRR8281218         0.00000000 0.0619555028      0.0000000   0.0155060204
## SRR8281219         0.10769382 0.0365928767      0.1552708   0.0320792458
## SRR8281226         0.03575150 0.4148672695      0.0000000   0.0000000000
## SRR8281236         0.01814658 0.0009746492      0.4032181   0.0265143088
## SRR8281230         0.09185887 0.0765481976      0.0000000   0.0007397341
## SRR8281233         0.12935356 0.3642042042      0.0000000   0.0146104309
## SRR8281244         0.20599738 0.0000000000      0.0000000   0.0111442172
## SRR8281245         0.03120120 0.2150193255      0.0000000   0.2086448744
## SRR8281243         0.04129983 0.3731328017      0.0000000   0.0712853735
## SRR8281251         0.01556184 0.1857497242      0.0000000   0.0649557029
## SRR8281238         0.13490224 0.1832561460      0.0000000   0.0058262672
## SRR8281250         0.05657334 0.2021661178      0.0000000   0.0026188791
##            Macrophages M2 Dendritic cells resting Dendritic cells activated
## SRR8281218      0.7049077            0.0000000000               0.000000000
## SRR8281219      1.1589577            0.0000000000               0.000000000
## SRR8281226      0.6474766            0.0000000000               0.004669831
## SRR8281236      1.0709844            0.0000000000               0.000000000
## SRR8281230      0.2730022            0.0005978992               0.000000000
## SRR8281233      2.1503686            0.0000000000               0.000000000
## SRR8281244      1.4746127            0.0000000000               0.000000000
## SRR8281245      1.4430364            0.0000000000               0.000000000
## SRR8281243      1.0892152            0.0000000000               0.000000000
## SRR8281251      0.7517625            0.0000000000               0.000000000
## SRR8281238      0.4755445            0.0004518878               0.000000000
## SRR8281250      0.4947339            0.0000000000               0.003493596
##            Mast cells resting Mast cells activated Eosinophils Neutrophils
## SRR8281218         0.11994310           0.00000000 0.003766691 0.000000000
## SRR8281219         0.24218289           0.00000000 0.000000000 0.079427915
## SRR8281226         0.01289667           0.00000000 0.002926060 0.123613285
## SRR8281236         0.06937423           0.00000000 0.000000000 0.017959545
## SRR8281230         0.24554128           0.00000000 0.000000000 0.055268474
## SRR8281233         0.00000000           0.04839376 0.000000000 0.008407325
## SRR8281244         0.00000000           0.21527869 0.000000000 0.028385274
## SRR8281245         0.21885148           0.00000000 0.000000000 0.040372665
## SRR8281243         0.13542538           0.00000000 0.000000000 0.006578541
## SRR8281251         0.00000000           0.20693722 0.001084478 0.058856146
## SRR8281238         0.00000000           0.32409274 0.013953275 0.047135836
## SRR8281250         0.00000000           0.10459983 0.000000000 0.035923878</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="Infiltration.html#cb70-1"></a><span class="co">#load plotfunction </span></span>
<span id="cb70-2"><a href="Infiltration.html#cb70-2"></a><span class="kw">source</span>(<span class="st">&quot;plotfunction.R&quot;</span>)</span>
<span id="cb70-3"><a href="Infiltration.html#cb70-3"></a></span>
<span id="cb70-4"><a href="Infiltration.html#cb70-4"></a><span class="co">#load metasheet for RIMA pipeline</span></span>
<span id="cb70-5"><a href="Infiltration.html#cb70-5"></a>metasheet &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;metasheet.csv&quot;</span>)</span>
<span id="cb70-6"><a href="Infiltration.html#cb70-6"></a></span>
<span id="cb70-7"><a href="Infiltration.html#cb70-7"></a><span class="co">#methods: CIBERSORT, quanTIseq, EPIC, MCP, xCell, TIMER</span></span>
<span id="cb70-8"><a href="Infiltration.html#cb70-8"></a><span class="kw">hmap</span>(<span class="dt">ta =</span> res_ciber, <span class="dt">meta =</span> metasheet, <span class="dt">methods =</span> <span class="st">&quot;CIBERSORT&quot;</span>, <span class="dt">phenotype =</span> <span class="st">&quot;Responder&quot;</span>)</span></code></pre></div>
<p><img src="RIMA-tutorial_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
</div>
<div id="box-plots-for-comparison-across-samples" class="section level3">
<h3><span class="header-section-number">6.7.3</span> Box plots for comparison across samples</h3>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="Infiltration.html#cb71-1"></a><span class="kw">boxfig</span>(<span class="dt">ta =</span> res_ciber, <span class="dt">meta =</span> metasheet, <span class="dt">methods =</span> <span class="st">&quot;CIBERSORT&quot;</span>, <span class="dt">phenotype =</span> <span class="st">&quot;Responder&quot;</span>)</span></code></pre></div>
<p><img src="RIMA-tutorial_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Repertoire.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Response.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
