<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Pre-processing of bulk RNA-seq data | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  



<meta name="date" content="2021-07-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="how-to-run-rima.html"/>
<link rel="next" href="Differential.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html"><i class="fa fa-check"></i><b>2</b> How to run RIMA</a><ul>
<li class="chapter" data-level="2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#install-rima-and-set-up-the-running-environment"><i class="fa fa-check"></i><b>2.1</b> Install RIMA and Set Up the Running Environment</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#download-pre-built-references"><i class="fa fa-check"></i><b>2.2</b> Download pre-built references</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#prepare-input-files"><i class="fa fa-check"></i><b>2.3</b> Prepare input files</a></li>
<li class="chapter" data-level="2.4" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#running-rima"><i class="fa fa-check"></i><b>2.4</b> Running RIMA</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#output"><i class="fa fa-check"></i><b>2.5</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-alignment"><i class="fa fa-check"></i><b>3.1</b> Read Alignment</a></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-without-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 without batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.2</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.3</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.4" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.4</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#bcr-cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> BCR cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-colonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Colonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
<li class="chapter" data-level="5.9" data-path="Repertoire.html"><a href="Repertoire.html#video-demo"><i class="fa fa-check"></i><b>5.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-cell-type"><i class="fa fa-check"></i><b>6.7</b> Comparison across cell type</a></li>
<li class="chapter" data-level="6.8" data-path="Infiltration.html"><a href="Infiltration.html#comparison-across-sample"><i class="fa fa-check"></i><b>6.8</b> Comparison across sample</a></li>
<li class="chapter" data-level="6.9" data-path="Infiltration.html"><a href="Infiltration.html#video-demo-1"><i class="fa fa-check"></i><b>6.9</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.1</b> TIDE</a><ul>
<li class="chapter" data-level="7.1.1" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.1.1</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.1.2" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.1.2</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.1.3" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.1.3</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.1.4" data-path="Response.html"><a href="Response.html#normaliztion"><i class="fa fa-check"></i><b>7.1.4</b> Normaliztion</a></li>
<li class="chapter" data-level="7.1.5" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.1.5</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.2</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.3</b> Response comparison analysis of biomarkers</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#video-demo-2"><i class="fa fa-check"></i><b>7.4</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Typing</a><ul>
<li class="chapter" data-level="8.1" data-path="HLA.html"><a href="HLA.html#identify-hla-type"><i class="fa fa-check"></i><b>8.1</b> Identify HLA type</a></li>
<li class="chapter" data-level="8.2" data-path="HLA.html"><a href="HLA.html#video-demo-3"><i class="fa fa-check"></i><b>8.2</b> Video demo</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>9</b> Fusion</a><ul>
<li class="chapter" data-level="9.1" data-path="fusion.html"><a href="fusion.html#fusion-calling"><i class="fa fa-check"></i><b>9.1</b> Fusion calling</a></li>
<li class="chapter" data-level="9.2" data-path="fusion.html"><a href="fusion.html#remove-fusion-homology"><i class="fa fa-check"></i><b>9.2</b> Remove fusion homology</a></li>
<li class="chapter" data-level="9.3" data-path="fusion.html"><a href="fusion.html#comparison-within-phenotype"><i class="fa fa-check"></i><b>9.3</b> Comparison within phenotype</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>10</b> Microbiome</a><ul>
<li class="chapter" data-level="10.1" data-path="microbiome.html"><a href="microbiome.html#centrifuge"><i class="fa fa-check"></i><b>10.1</b> Centrifuge</a></li>
<li class="chapter" data-level="10.2" data-path="microbiome.html"><a href="microbiome.html#microial-classfication-from-rna-seq-data"><i class="fa fa-check"></i><b>10.2</b> Microial classfication from RNA-seq data</a></li>
<li class="chapter" data-level="10.3" data-path="microbiome.html"><a href="microbiome.html#video-demo-4"><i class="fa fa-check"></i><b>10.3</b> Video demo</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Preprocessing" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Pre-processing of bulk RNA-seq data</h1>
<p>In this chapter, we will introduce RIMA align RNA-seq data, check the data quality, quantify gene expression and handle batch effects across samples. To run RIMA preprocess module, in <strong>execution.yaml</strong>, set preprocess module to <em>true</em>. The individual run of preprocess includes read alignment, quality check and gene quantification. The cohort run of preprocess module will merge alignment report, quality metrics, gene quantification and conduct batch effect removal.</p>
<pre><code>## execution.yaml ##
preprocess_individual: true
preprocess_cohort: true</code></pre>
<div id="read-alignment" class="section level2">
<h2><span class="header-section-number">3.1</span> Read Alignment</h2>
<p><strong>Align reads with STAR</strong></p>
<p><a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">STAR</a> is one of the most common tools used for bulk RNA-seq data alignment to generate transcriptome BAM or genomic BAM output, which can be downloaded at <a href="https://github.com/alexdobin/STAR"> here</a>. A tutorial for STAR is available <a href="https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html">here</a>.</p>
<p>When using STAR, the first step is to create genome index. In our RIMA pipeline, we download the human genome (hg38) STAR index from <a href="https://gdc.cancer.gov/">Genomic Data Commons (GDC) </a>.Let’s use sample <em>SRR8281218</em> as an example.</p>
<p>RIMA runs STAR using the following command structure:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="Preprocessing.html#cb16-1"></a>      STAR <span class="op">--</span>runThreadN <span class="dv">4</span> <span class="op">--</span>genomeDir .<span class="op">/</span>ref_files<span class="op">/</span>v22</span>
<span id="cb16-2"><a href="Preprocessing.html#cb16-2"></a>        <span class="op">--</span>outReadsUnmapped None </span>
<span id="cb16-3"><a href="Preprocessing.html#cb16-3"></a>        <span class="op">--</span>chimSegmentMin <span class="dv">12</span> </span>
<span id="cb16-4"><a href="Preprocessing.html#cb16-4"></a>        <span class="op">--</span>chimJunctionOverhangMin <span class="dv">12</span> </span>
<span id="cb16-5"><a href="Preprocessing.html#cb16-5"></a>        <span class="op">--</span>chimOutJunctionFormat <span class="dv">1</span> </span>
<span id="cb16-6"><a href="Preprocessing.html#cb16-6"></a>        <span class="op">--</span>alignSJDBoverhangMin <span class="dv">10</span> </span>
<span id="cb16-7"><a href="Preprocessing.html#cb16-7"></a>        <span class="op">--</span>alignMatesGapMax <span class="dv">1000000</span> </span>
<span id="cb16-8"><a href="Preprocessing.html#cb16-8"></a>        <span class="op">--</span>alignIntronMax <span class="dv">1000000</span> </span>
<span id="cb16-9"><a href="Preprocessing.html#cb16-9"></a>        <span class="op">--</span>alignSJstitchMismatchNmax <span class="dv">5</span> <span class="dv">-1</span> <span class="dv">5</span> <span class="dv">5</span> </span>
<span id="cb16-10"><a href="Preprocessing.html#cb16-10"></a>        <span class="op">--</span>outSAMstrandField intronMotif </span>
<span id="cb16-11"><a href="Preprocessing.html#cb16-11"></a>        <span class="op">--</span>outSAMunmapped Within </span>
<span id="cb16-12"><a href="Preprocessing.html#cb16-12"></a>        <span class="op">--</span>outSAMtype BAM Unsorted </span>
<span id="cb16-13"><a href="Preprocessing.html#cb16-13"></a>        <span class="op">--</span>readFilesIn {input} </span>
<span id="cb16-14"><a href="Preprocessing.html#cb16-14"></a>        <span class="op">--</span>chimMultimapScoreRange <span class="dv">10</span> </span>
<span id="cb16-15"><a href="Preprocessing.html#cb16-15"></a>        <span class="op">--</span>chimMultimapNmax <span class="dv">10</span> </span>
<span id="cb16-16"><a href="Preprocessing.html#cb16-16"></a>        <span class="op">--</span>chimNonchimScoreDropMin <span class="dv">10</span> </span>
<span id="cb16-17"><a href="Preprocessing.html#cb16-17"></a>        <span class="op">--</span>peOverlapNbasesMin <span class="dv">12</span> </span>
<span id="cb16-18"><a href="Preprocessing.html#cb16-18"></a>        <span class="op">--</span>peOverlapMMp <span class="fl">0.1</span> </span>
<span id="cb16-19"><a href="Preprocessing.html#cb16-19"></a>        <span class="op">--</span>genomeLoad NoSharedMemory </span>
<span id="cb16-20"><a href="Preprocessing.html#cb16-20"></a>        <span class="op">--</span>outSAMheaderHD <span class="op">@</span>HD VN<span class="op">:</span><span class="fl">1.4</span></span>
<span id="cb16-21"><a href="Preprocessing.html#cb16-21"></a>        <span class="op">--</span>outFileNamePrefix {params.prefix}</span>
<span id="cb16-22"><a href="Preprocessing.html#cb16-22"></a>        <span class="op">--</span>quantMode TranscriptomeSAM</span></code></pre></div>
<p>RIMA renamed read aligment outputs according to sample id.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="Preprocessing.html#cb17-1"></a>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.sorted.bam             </span>
<span id="cb17-2"><a href="Preprocessing.html#cb17-2"></a>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.transcriptome.bam      </span>
<span id="cb17-3"><a href="Preprocessing.html#cb17-3"></a>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.Chimeric.out.junction </span>
<span id="cb17-4"><a href="Preprocessing.html#cb17-4"></a>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.Log.final.out</span></code></pre></div>
<p>RIMA uses samtools to generate statistics from the alignment BAM file:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="Preprocessing.html#cb18-1"></a>samtools stats analysis<span class="op">/</span>star<span class="op">/</span>SRR828121<span class="op">/</span>SRR8281218.sorted.bam <span class="op">|</span><span class="st"> </span>grep <span class="op">^</span>SN <span class="op">|</span><span class="st"> </span>cut <span class="op">-</span>f <span class="dv">2</span><span class="op">-</span><span class="st"> </span><span class="er">&gt;</span><span class="st"> </span>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.sorted.bam.stat.txt</span></code></pre></div>
<p><strong>Merging STAR alignment report</strong></p>
<p>After individual alignment run of each sample, RIMA will merge the alignment report named <em>STAR_Align_Report.csv</em> from each sample. Below is an example the merged output:</p>
<pre><code>STAR,SRR8281218,SRR8281219,SRR8281226,SRR8281236,SRR8281230,SRR8281233,SRR8281244,SRR8281245,SRR8281243,SRR8281251,SRR8281238,SRR8281250
Number_of_input_reads,31058961,30244952,30144822,30214970,30219317,30294851,30233324,30188854,30267936,30155916,30238326,30205607
Average_input_read_length,200,200,200,200,200,200,200,200,200,200,200,200
Uniquely_mapped_reads_number,29396418,28842972,27785223,28625526,28552141,29084522,29172831,28527204,28611921,28526789,26648329,28820905
Uniquely_mapped_reads_%,94.65%,95.36%,92.17%,94.74%,94.48%,96.00%,96.49%,94.50%,94.53%,94.60%,88.13%,95.42%
Average_mapped_length,198.82,198.69,198.65,198.64,198.64,198.54,198.64,198.54,198.66,198.64,198.71,198.62
Number_of_splices:_Total,14076881,16545751,18439209,15091999,14432943,15290737,16116559,18614840,14065847,13967615,9829323,14650425
Number_of_splices:_Annotated_(sjdb),14043726,16506835,18399706,15050436,14388389,15246382,16073843,18570667,14031241,13930154,9801357,14610503
Number_of_splices:_GT/AG,13944521,16380126,18275777,14941222,14287227,15142625,15961566,18425036,13939221,13836881,9738054,14508837
Number_of_splices:_GC/AG,106767,133082,130042,118569,116171,114096,121618,134112,97534,102169,69796,112736
Number_of_splices:_AT/AC,11811,14116,14898,13200,12486,11590,12575,12673,11867,11783,7031,13252
Number_of_splices:_Non-canonical,13782,18427,18492,19008,17059,22426,20800,43019,17225,16782,14442,15600
Mismatch_rate_per_base_%,0.20%,0.18%,0.17%,0.18%,0.19%,0.18%,0.17%,0.19%,0.19%,0.19%,0.18%,0.17%
Deletion_rate_per_base,0.01%,0.00%,0.00%,0.01%,0.00%,0.01%,0.00%,0.00%,0.01%,0.01%,0.01%,0.00%
Deletion_average_length,1.29,1.29,1.26,1.29,1.28,1.21,1.29,1.26,1.22,1.22,1.12,1.23
Insertion_rate_per_base,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%,0.01%
Insertion_average_length,1.40,1.43,1.42,1.44,1.44,1.45,1.42,1.41,1.41,1.40,1.37,1.40
Number_of_reads_mapped_to_multiple_loci,1331220,1119545,2078090,1280274,1347073,901525,766599,1332933,1325151,1337251,3322861,1103343
%_of_reads_mapped_to_multiple_loci,4.29%,3.70%,6.89%,4.24%,4.46%,2.98%,2.54%,4.42%,4.38%,4.43%,10.99%,3.65%
Number_of_reads_mapped_to_too_many_loci,9535,10773,11999,12262,15652,9758,8344,10240,8594,10179,9217,8381
%_of_reads_mapped_to_too_many_loci,0.03%,0.04%,0.04%,0.04%,0.05%,0.03%,0.03%,0.03%,0.03%,0.03%,0.03%,0.03%
%_of_reads_unmapped:_too_many_mismatches,0.08%,0.09%,0.12%,0.13%,0.13%,0.14%,0.10%,0.24%,0.13%,0.12%,0.11%,0.09%
%_of_reads_unmapped:_too_short,0.92%,0.76%,0.73%,0.81%,0.83%,0.81%,0.80%,0.77%,0.90%,0.77%,0.70%,0.78%
%_of_reads_unmapped:_other,0.04%,0.04%,0.05%,0.04%,0.05%,0.03%,0.04%,0.04%,0.03%,0.05%,0.04%,0.04%
Number_of_chimeric_reads,592271,594198,549596,631747,635307,717293,631225,666748,637640,622475,600743,636839
%_of_chimeric_reads,1.91%,1.96%,1.82%,2.09%,2.10%,2.37%,2.09%,2.21%,2.11%,2.06%,1.99%,2.11%</code></pre>
</div>
<div id="quality-control" class="section level2">
<h2><span class="header-section-number">3.2</span> Quality Control</h2>
<p><strong>Downsampling alignment BAMs</strong></p>
<p>RseQC is a standard tool for checking the quality of reads alignment, providing the principal measurements of RNA-seq data quality. An RseQC tutorial is available <a href="http://rseqc.sourceforge.net/">here</a>.<br />
</p>
<p>To reduce running time, we first sub-sampled sorted BAM files to be used by RseQC to assess alignment quality. If parameter ‘rseqc_ref’ is set to ‘house_keeping’, RIMA will further extract alignments of house keeping genes.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="Preprocessing.html#cb20-1"></a><span class="co">## Example of subsampling 50% of input alignment</span></span>
<span id="cb20-2"><a href="Preprocessing.html#cb20-2"></a>samtools view <span class="op">-</span>s <span class="fl">0.5</span> <span class="op">-</span>b analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218.sorted.bam  <span class="op">&gt;</span><span class="st"> </span>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218_downsampling.bam </span>
<span id="cb20-3"><a href="Preprocessing.html#cb20-3"></a></span>
<span id="cb20-4"><a href="Preprocessing.html#cb20-4"></a><span class="co"># Extract the alignment of housekeeping gene.</span></span>
<span id="cb20-5"><a href="Preprocessing.html#cb20-5"></a>bedtools intersect <span class="op">-</span>a analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218_downsampling.bam  <span class="op">-</span>b .<span class="op">/</span>ref_files<span class="op">/</span>housekeeping_refseqGenes.bed <span class="op">&gt;</span><span class="st"> </span>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218_downsampling_housekeeping.bam </span>
<span id="cb20-6"><a href="Preprocessing.html#cb20-6"></a></span>
<span id="cb20-7"><a href="Preprocessing.html#cb20-7"></a><span class="co"># index BAM</span></span>
<span id="cb20-8"><a href="Preprocessing.html#cb20-8"></a>samtools index analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218_downsampling_housekeeping.bam  <span class="op">&gt;</span><span class="st"> </span>analysis<span class="op">/</span>star<span class="op">/</span>SRR8281218<span class="op">/</span>SRR8281218_downsampling_housekeeping.bam.bai</span></code></pre></div>
<p><strong>RseQC Quality metrics</strong></p>
<ol style="list-style-type: decimal">
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0922-z#Sec2">Transcript integrity number (TIN)</a> is the most widely used measure of RNA integrity, which is the percentage of transcripts that have uniform read coverage across the genome. RseQC calculates the TIN of each transcript and reports mean TIN, median TIN and standard deviation for all transcripts in a sample. The <a href="http://rseqc.sourceforge.net/#tin-py">median TIN score (medTIN) </a> across all transcripts is commonly used to indicate the RNA integrity of each sample.<br />
</li>
<li><a href="http://rseqc.sourceforge.net/#read-distribution-py">Read distribution </a> summarizes the fraction of reads aligned into different genomic regions, such as exon and intron regions.<br />
</li>
<li><a href="http://rseqc.sourceforge.net/#genebody-coverage-py">Gene body coverage</a> shows the RNA-seq read coverage over the gene body.<br />
</li>
</ol>
<p>Low quality samples have low alignment fractions, low integrity (median TIN) and/or abnormal read distributions and should be removed in downstream analysis.</p>
<p><strong>Merge quality metrics</strong></p>
<p>RIMA will also merge quality metrics for all samples. Below is the merged outputs for TIN score <em>tin_score_summary.txt</em> and read distribution <em>read_distrib.matrix.tab</em>:</p>
<pre><code>##tin_score_summary.txt##
Bam_file    TIN(mean)   TIN(median) TIN(stdev)
SRR8281218_downsampling_housekeeping.bam    72.59011823410742   76.61544321066245   15.70457852012789
SRR8281219_downsampling_housekeeping.bam    74.53459359515176   78.36139111185804   14.862696392336295
SRR8281226_downsampling_housekeeping.bam    72.12569765806015   76.04482057429738   15.150628049210777
SRR8281236_downsampling_housekeeping.bam    75.10959004809101   79.0211107295296    14.112536675463657
SRR8281230_downsampling_housekeeping.bam    74.95210786264646   78.18898561188249   14.084506430648968
SRR8281233_downsampling_housekeeping.bam    71.91235332638948   76.36385765230236   16.193874225558574
SRR8281244_downsampling_housekeeping.bam    72.42514699067206   76.9844608754321    16.270160903196263
SRR8281245_downsampling_housekeeping.bam    72.16726184355547   76.41674529483181   15.380595156507917
SRR8281243_downsampling_housekeeping.bam    71.56229409539716   74.7019940049169    13.711275796768888
SRR8281251_downsampling_housekeeping.bam    73.02855865353006   77.30010010150008   14.902687257461483
SRR8281238_downsampling_housekeeping.bam    66.23444336304713   70.08838336433544   17.437545601110816
SRR8281250_downsampling_housekeeping.bam    75.25874145796432   78.83524471249612   14.125154826821316

##read_distrib.matrix.tab##
Feature SRR8281218  SRR8281219  SRR8281226  SRR8281236  SRR8281230  SRR8281233  SRR8281244  SRR8281245  SRR8281243  SRR8281251  SRR82
81238   SRR8281250
TES_down_1kb    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00
TSS_up_1kb  0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00
TSS_up_5kb  0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00
Introns 0.05    0.05    0.03    0.06    0.08    0.06    0.05    0.04    0.04    0.04    0.04    0.04
TES_down_5kb    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00
CDS_Exons   0.80    0.81    0.85    0.79    0.77    0.79    0.79    0.83    0.80    0.80    0.81    0.80
3&#39;UTR_Exons 0.12    0.11    0.08    0.12    0.11    0.12    0.11    0.09    0.12    0.11    0.11    0.12
TES_down_10kb   0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00
5&#39;UTR_Exons 0.04    0.04    0.04    0.03    0.03    0.03    0.04    0.04    0.03    0.04    0.04    0.04
TSS_up_10kb 0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00    0.00</code></pre>
Cohort level gene body coverage are plotted together. Below is an example of <em>geneBodyCoverage.curves.png</em>:
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-7"></span>
<img src="images/geneBodyCoverage.curves.png" alt="Gene Body Coverage Curves"  />
<p class="caption">
Figure 3.1: Gene Body Coverage Curves
</p>
</div>
</div>
<div id="gene-quantification" class="section level2">
<h2><span class="header-section-number">3.3</span> Gene quantification</h2>
<p>After the alignment of sequencing reads and quality check, we quantify the gene or transcript expression from the BAM files.Both <a href="https://combine-lab.github.io/salmon/getting_started/">Salmon</a> and <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323">RSEM</a> are widely used for gene quantification. Salmon conducts fast transcript-level quantification, and RSEM performs both gene-level and transcript-level quantification. Salmon is usually faster and less memory-consuming than RSEM. Both tools generate transcripts per million (TPM), reads per kilobase of exon model (transcript) per million mapped reads (RPKM) and fragments per kilobase of exon model (transcript) per million mapped reads (FPKM) expression. All three measures (TPM, RPKM, FPKM) are normalization methods that account for both sequencing depth and gene lengths. (Longer transcripts will have more reads.) TPM is currently thought to be the best method to normalize gene counts. <a href="https://htseq.readthedocs.io/en/master/tour.html#counting-reads-by-genes">HTSeq </a> and STAR quantify gene expression according to mapped read counts. Both normalized gene expression and read counts can be used for differential expression analysis.<br />
</p>
<p>RIMA used Salmon for gene quantification and input for Salmon is the transcriptome alignment result:</p>
<pre><code>salmon quant -t ./ref_files/salmon_gdc_index/gencode.v22.ts.fa -l A -a analysis/star/SRR8281218/SRR8281218.transcriptome.bam  -o analysis/salmon/SRR8281218</code></pre>
<p>The following is an example of output file from Salmon</p>
<pre><code>Name    Length  EffectiveLength TPM NumReads
ENST00000456328.2   1657    1474.085    0.000000    0.000
ENST00000450305.2   632 449.595 0.000000    0.000
ENST00000488147.1   1351    1168.085    7.412619    139.959
ENST00000619216.1   68  69.000  0.000000    0.000
ENST00000473358.1   712 529.382 0.000000    0.000
ENST00000469289.1   535 353.431 0.000000    0.000
ENST00000607096.1   138 16.202  0.000000    0.000</code></pre>
<p>We then combined the TPMs at gene level use ‘txtimport’ from DESeq2 R packages and results will be:</p>
<pre><code>## First 10 rows of merge TPMs ##
SRR8281218,SRR8281219,SRR8281226,SRR8281236,SRR8281233,SRR8281230,SRR8281244,SRR8281245,SRR8281243,SRR8281238,SRR8281251,SRR8281250
5_8S_rRNA,0,0,0,0,0,0,0,0,0,0,0,0
5S_rRNA,0,0,0,0,0,0,0,0,0,0,0,0
7SK,0,0.75,1,5.864,5.667,5,1.961,2.67,10.47,3.531,5,2.981
A1BG,98.378,248.265,222.032,106.318,159.868,262.034,210.18,157.196,83.865,137.558,130.693,193.207
A1BG-AS1,82.273,252.763,70.572,80.758,100.73,297.316,119.662,145.835,66.107,59.74,90.452,168.508
A1CF,2,2,0,3,0,0,0,0,1,2,1,196
A2M,14326.273,9018.603,27465.295,22061.332,31646.181,4559.962,14865.119,20065.994,30592.578,19598.148,12355.795,7203.953
A2M-AS1,59.728,6.397,12.704,36.667,52.819,4.038,24.881,13.006,34.423,23.852,6.206,25.047
A2ML1,180.001,0,17,42,83,55,93,0,12,10,9,763.998
A2ML1-AS1,0,0,0,0,0,0,0,0,0,3.506,0,0</code></pre>
</div>
<div id="batch-effect-removal" class="section level2">
<h2><span class="header-section-number">3.4</span> Batch effect removal</h2>
<p>Batch effects across samples are easily ignored but worth considering for immunotherapy cohort studies. Bacth effects are usually caused by unbalanced experimental design and confound the estimation of group differences. For example, samples processed at different facilities may have sequencing differences that are not due to actual biological differences in the samples themselves. To avoid confounding actual biology variation with the effects of experimental design, <a href="https://www.degruyter.com/document/doi/10.2202/1544-6115.1027/html"> limma </a> and <a href="https://academic.oup.com/biostatistics/article/8/1/118/252073?login=true">Combat</a> are common approaches to correct batch effects. Limma uses a two-way ANOVA approach. Combat uses an empirical Bayes approach, which is critical for small batches to avoid over-correction. For large batches, both methods should be similar. The <a href="https://bioconductor.org/packages/release/bioc/html/sva.html">sva</a> R package integrates both methods for batch effect correction.
Principal components analysis (PCA) or unsupervised clustering before and after batch effect removal is an excellent way to validate that a batch effect has been removed.</p>
<p>To evaluate if your samples have a batch effect, RIMA will generate PCA plots of gene expression data before and after batch effect removal by limma. To utilize this feature, modify the “batch” parameter in the config.yaml file for your run. No batch effects found in those 6 samples so we manually added ‘syn_batch’ column in the metasheet for demo purpose.</p>
Example of PCA before and after batch correction using limma.
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-8"></span>
<img src="images/PCA_synbatch.png" alt="Before(left) and after(right) batch correction with Limma"  />
<p class="caption">
Figure 3.2: Before(left) and after(right) batch correction with Limma
</p>
</div>
</div>
<div id="video-demo-of-rima" class="section level2">
<h2><span class="header-section-number">3.5</span> Video demo of RIMA</h2>
<iframe width="560" height="315" src="https://www.ynutube.com/embed/NEbi-pYNhak" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><br />
</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="how-to-run-rima.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Differential.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
