<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Immune repertoire | Tutorial of RNA-seq tumor immunity analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Yang Liu, Jennifer Altreuter, Yang Lin" />


<meta name="date" content="2022-12-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Differential.html"/>
<link rel="next" href="Infiltration.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html"><i class="fa fa-check"></i><b>2</b> How to run RIMA</a><ul>
<li class="chapter" data-level="2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#install-rima-and-set-up-the-running-environment"><i class="fa fa-check"></i><b>2.1</b> Install RIMA and Set Up the Running Environment</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#download-pre-built-references"><i class="fa fa-check"></i><b>2.2</b> Download pre-built references</a><ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial"><i class="fa fa-check"></i><b>2.2.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#prepare-input-files"><i class="fa fa-check"></i><b>2.3</b> Prepare input files</a><ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#video-tutorial-1"><i class="fa fa-check"></i><b>2.3.1</b> Video tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#running-rima"><i class="fa fa-check"></i><b>2.4</b> Running RIMA</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-run-rima.html"><a href="how-to-run-rima.html#output"><i class="fa fa-check"></i><b>2.5</b> Output</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="Preprocessing.html"><a href="Preprocessing.html"><i class="fa fa-check"></i><b>3</b> Pre-processing of bulk RNA-seq data</a><ul>
<li class="chapter" data-level="3.1" data-path="Preprocessing.html"><a href="Preprocessing.html#read-alignment"><i class="fa fa-check"></i><b>3.1</b> Read Alignment</a></li>
<li class="chapter" data-level="3.2" data-path="Preprocessing.html"><a href="Preprocessing.html#quality-control"><i class="fa fa-check"></i><b>3.2</b> Quality Control</a></li>
<li class="chapter" data-level="3.3" data-path="Preprocessing.html"><a href="Preprocessing.html#gene-quantification"><i class="fa fa-check"></i><b>3.3</b> Gene quantification</a></li>
<li class="chapter" data-level="3.4" data-path="Preprocessing.html"><a href="Preprocessing.html#batch-effect-removal"><i class="fa fa-check"></i><b>3.4</b> Batch effect removal</a><ul>
<li class="chapter" data-level="3.4.1" data-path="Preprocessing.html"><a href="Preprocessing.html#start-with-tpm-matrix"><i class="fa fa-check"></i><b>3.4.1</b> Start with TPM matrix</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="Preprocessing.html"><a href="Preprocessing.html#video-demo-of-rima"><i class="fa fa-check"></i><b>3.5</b> Video demo of RIMA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Differential.html"><a href="Differential.html"><i class="fa fa-check"></i><b>4</b> Differential gene analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="Differential.html"><a href="Differential.html#differential-gene-expression-using-deseq2"><i class="fa fa-check"></i><b>4.1</b> Differential gene expression using DESeq2</a><ul>
<li class="chapter" data-level="4.1.1" data-path="Differential.html"><a href="Differential.html#inputs-of-deseq2"><i class="fa fa-check"></i><b>4.1.1</b> Inputs of DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="Differential.html"><a href="Differential.html#different-gene-expression-analysis"><i class="fa fa-check"></i><b>4.2</b> Different gene expression analysis</a><ul>
<li class="chapter" data-level="4.2.1" data-path="Differential.html"><a href="Differential.html#running-deseq2-without-batch-effect"><i class="fa fa-check"></i><b>4.2.1</b> Running DESeq2 without batch effect</a></li>
<li class="chapter" data-level="4.2.2" data-path="Differential.html"><a href="Differential.html#running-deseq2-with-batch-effect"><i class="fa fa-check"></i><b>4.2.2</b> Running DESeq2 with batch effect</a></li>
<li class="chapter" data-level="4.2.3" data-path="Differential.html"><a href="Differential.html#output-from-deseq2"><i class="fa fa-check"></i><b>4.2.3</b> Output from DESeq2</a></li>
<li class="chapter" data-level="4.2.4" data-path="Differential.html"><a href="Differential.html#volcano-plots"><i class="fa fa-check"></i><b>4.2.4</b> Volcano Plot(s)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Differential.html"><a href="Differential.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>4.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="4.4" data-path="Differential.html"><a href="Differential.html#single-sample-gene-set-enrichment-analysis-ssgsea"><i class="fa fa-check"></i><b>4.4</b> Single sample gene set enrichment analysis (ssGSEA)</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Repertoire.html"><a href="Repertoire.html"><i class="fa fa-check"></i><b>5</b> Immune repertoire</a><ul>
<li class="chapter" data-level="5.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4"><i class="fa fa-check"></i><b>5.1</b> TRUST4</a><ul>
<li class="chapter" data-level="5.1.1" data-path="Repertoire.html"><a href="Repertoire.html#trust4-output"><i class="fa fa-check"></i><b>5.1.1</b> TRUST4 output</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="Repertoire.html"><a href="Repertoire.html#cluster-calculation"><i class="fa fa-check"></i><b>5.2</b> Cluster calculation</a></li>
<li class="chapter" data-level="5.3" data-path="Repertoire.html"><a href="Repertoire.html#clonotypes-per-kilo-reads-cpk"><i class="fa fa-check"></i><b>5.3</b> Clonotypes per kilo-reads (CPK)</a></li>
<li class="chapter" data-level="5.4" data-path="Repertoire.html"><a href="Repertoire.html#tcr-and-bcr-entropy-and-clonality"><i class="fa fa-check"></i><b>5.4</b> TCR and BCR entropy and Clonality</a></li>
<li class="chapter" data-level="5.5" data-path="Repertoire.html"><a href="Repertoire.html#fraction-of-reads-mapped-to-tcr-and-bcr"><i class="fa fa-check"></i><b>5.5</b> Fraction of reads mapped to TCR and BCR</a></li>
<li class="chapter" data-level="5.6" data-path="Repertoire.html"><a href="Repertoire.html#somatic-hypermutation-rate-of-bcr"><i class="fa fa-check"></i><b>5.6</b> Somatic hypermutation rate of BCR</a></li>
<li class="chapter" data-level="5.7" data-path="Repertoire.html"><a href="Repertoire.html#ig-isotype-frequency"><i class="fa fa-check"></i><b>5.7</b> Ig isotype frequency</a></li>
<li class="chapter" data-level="5.8" data-path="Repertoire.html"><a href="Repertoire.html#cohort-analysis"><i class="fa fa-check"></i><b>5.8</b> Cohort analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Infiltration.html"><a href="Infiltration.html"><i class="fa fa-check"></i><b>6</b> Immune Infiltration</a><ul>
<li class="chapter" data-level="6.1" data-path="Infiltration.html"><a href="Infiltration.html#cibersort"><i class="fa fa-check"></i><b>6.1</b> Cibersort</a></li>
<li class="chapter" data-level="6.2" data-path="Infiltration.html"><a href="Infiltration.html#timer"><i class="fa fa-check"></i><b>6.2</b> TIMER</a></li>
<li class="chapter" data-level="6.3" data-path="Infiltration.html"><a href="Infiltration.html#quantiseq"><i class="fa fa-check"></i><b>6.3</b> quanTIseq</a></li>
<li class="chapter" data-level="6.4" data-path="Infiltration.html"><a href="Infiltration.html#xcell"><i class="fa fa-check"></i><b>6.4</b> xCell</a></li>
<li class="chapter" data-level="6.5" data-path="Infiltration.html"><a href="Infiltration.html#epic"><i class="fa fa-check"></i><b>6.5</b> EPIC</a></li>
<li class="chapter" data-level="6.6" data-path="Infiltration.html"><a href="Infiltration.html#mcp-counter"><i class="fa fa-check"></i><b>6.6</b> MCP-counter</a></li>
<li class="chapter" data-level="6.7" data-path="Infiltration.html"><a href="Infiltration.html#starting-with-tpm-matrix"><i class="fa fa-check"></i><b>6.7</b> Starting with TPM matrix</a><ul>
<li class="chapter" data-level="6.7.1" data-path="Infiltration.html"><a href="Infiltration.html#correlation-plots"><i class="fa fa-check"></i><b>6.7.1</b> Correlation Plots</a></li>
<li class="chapter" data-level="6.7.2" data-path="Infiltration.html"><a href="Infiltration.html#heatmaps-for-comparisons-across-cell-types"><i class="fa fa-check"></i><b>6.7.2</b> Heatmaps for comparisons across cell types</a></li>
<li class="chapter" data-level="6.7.3" data-path="Infiltration.html"><a href="Infiltration.html#box-plots-for-comparison-across-samples"><i class="fa fa-check"></i><b>6.7.3</b> Box plots for comparison across samples</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Response.html"><a href="Response.html"><i class="fa fa-check"></i><b>7</b> Immune Response</a><ul>
<li class="chapter" data-level="7.1" data-path="Response.html"><a href="Response.html#characterizing-the-tme-and-t-cell-functionality"><i class="fa fa-check"></i><b>7.1</b> Characterizing the TME and T cell functionality</a></li>
<li class="chapter" data-level="7.2" data-path="Response.html"><a href="Response.html#tide"><i class="fa fa-check"></i><b>7.2</b> TIDE</a><ul>
<li class="chapter" data-level="7.2.1" data-path="Response.html"><a href="Response.html#config.yaml-sections-relevant-to-tide"><i class="fa fa-check"></i><b>7.2.1</b> Config.yaml sections relevant to TIDE</a></li>
<li class="chapter" data-level="7.2.2" data-path="Response.html"><a href="Response.html#cold-and-hot-tumors"><i class="fa fa-check"></i><b>7.2.2</b> Cold and hot tumors</a></li>
<li class="chapter" data-level="7.2.3" data-path="Response.html"><a href="Response.html#t-cell-dysfunction"><i class="fa fa-check"></i><b>7.2.3</b> T cell dysfunction</a></li>
<li class="chapter" data-level="7.2.4" data-path="Response.html"><a href="Response.html#t-cell-exclusion"><i class="fa fa-check"></i><b>7.2.4</b> T cell exclusion</a></li>
<li class="chapter" data-level="7.2.5" data-path="Response.html"><a href="Response.html#normalization"><i class="fa fa-check"></i><b>7.2.5</b> Normalization</a></li>
<li class="chapter" data-level="7.2.6" data-path="Response.html"><a href="Response.html#running-tide"><i class="fa fa-check"></i><b>7.2.6</b> Running TIDE</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Response.html"><a href="Response.html#microsatelite-instability-msi"><i class="fa fa-check"></i><b>7.3</b> Microsatelite instability (MSI)</a></li>
<li class="chapter" data-level="7.4" data-path="Response.html"><a href="Response.html#response-comparison-analysis-of-biomarkers"><i class="fa fa-check"></i><b>7.4</b> Response comparison analysis of biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="HLA.html"><a href="HLA.html"><i class="fa fa-check"></i><b>8</b> HLA Typing &amp; Neoantigens</a><ul>
<li class="chapter" data-level="8.1" data-path="HLA.html"><a href="HLA.html#identify-hla-type"><i class="fa fa-check"></i><b>8.1</b> Identify HLA type</a></li>
<li class="chapter" data-level="8.2" data-path="HLA.html"><a href="HLA.html#hla-oncoplot"><i class="fa fa-check"></i><b>8.2</b> HLA Oncoplot</a></li>
<li class="chapter" data-level="8.3" data-path="HLA.html"><a href="HLA.html#neoantigen-identification"><i class="fa fa-check"></i><b>8.3</b> Neoantigen Identification</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="fusion.html"><a href="fusion.html"><i class="fa fa-check"></i><b>9</b> Fusion</a><ul>
<li class="chapter" data-level="9.1" data-path="fusion.html"><a href="fusion.html#fusion-calling"><i class="fa fa-check"></i><b>9.1</b> Fusion calling</a></li>
<li class="chapter" data-level="9.2" data-path="fusion.html"><a href="fusion.html#remove-fusion-homology"><i class="fa fa-check"></i><b>9.2</b> Remove fusion homology</a></li>
<li class="chapter" data-level="9.3" data-path="fusion.html"><a href="fusion.html#comparison-within-phenotype"><i class="fa fa-check"></i><b>9.3</b> Comparison within phenotype</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="microbiome.html"><a href="microbiome.html"><i class="fa fa-check"></i><b>10</b> Microbiome</a><ul>
<li class="chapter" data-level="10.1" data-path="microbiome.html"><a href="microbiome.html#centrifuge"><i class="fa fa-check"></i><b>10.1</b> Centrifuge</a></li>
<li class="chapter" data-level="10.2" data-path="microbiome.html"><a href="microbiome.html#microbial-classfication-from-rna-seq-data"><i class="fa fa-check"></i><b>10.2</b> Microbial classfication from RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html"><i class="fa fa-check"></i><b>11</b> Customize your own reference</a><ul>
<li class="chapter" data-level="11.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#reference-fasta"><i class="fa fa-check"></i><b>11.1</b> Reference fasta</a></li>
<li class="chapter" data-level="11.2" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gene-annotation-file-gtf"><i class="fa fa-check"></i><b>11.2</b> Gene annotation file (gtf)</a></li>
<li class="chapter" data-level="11.3" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-star-index"><i class="fa fa-check"></i><b>11.3</b> build STAR index</a></li>
<li class="chapter" data-level="11.4" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#rseqc-reference-files"><i class="fa fa-check"></i><b>11.4</b> RSeQC reference files</a></li>
<li class="chapter" data-level="11.5" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#build-salmon-index"><i class="fa fa-check"></i><b>11.5</b> build salmon index</a><ul>
<li class="chapter" data-level="11.5.1" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#gmt-file-for-gene-set-analysis"><i class="fa fa-check"></i><b>11.5.1</b> GMT file for gene set analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#star-fusion-genome-resource-lib"><i class="fa fa-check"></i><b>11.6</b> STAR-Fusion genome resource lib</a></li>
<li class="chapter" data-level="11.7" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#centrifuge-index"><i class="fa fa-check"></i><b>11.7</b> Centrifuge index</a></li>
<li class="chapter" data-level="11.8" data-path="customize-your-own-reference.html"><a href="customize-your-own-reference.html#trust4-reference-files"><i class="fa fa-check"></i><b>11.8</b> TRUST4 reference files</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial of RNA-seq tumor immunity analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Repertoire" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Immune repertoire</h1>
<p>A key part of immune specific analysis is assessing the nature of the T and B cells that make up the tumor microenvironment. This assessment includes, but is not limited to, measures such as receptor diversity and clonality.</p>
<p>Once tumors are infiltrated with T cells and B cells, the T cells and B cells are activated if their receptors – TCRs and BCRs, respectively – recognize and bind to tumor-associated antigens. TCRs and BCRs are highly diverse to allow different T and B cells to recognize different pathogens or tumor-associated antigens. Part of the diversity of TCRs and BCRs is due to V(D)J recombination which mixes and matches different V, D and J segments of the genome in order to produce a wide variety of receptors.</p>
<p>Complementary-determining region 3 (<strong>CDR3</strong>) is a highly variable region of both TCRs and BCRs that is central to the antigen binding site of these receptors. V(D)J junctional sequence assembly is a major source of TCR/BCR CDR3 diversity. <strong>Immune repertoire profiling</strong> (characterizing the CDR3 sequences of TCRs and BCRs in a tumor sample) is essential for quantifying T/B cell diversity and clonality. These measures, in turn, are important tumor immune characteristics and key indicators of immunotherapy response.</p>
<p>Currently, newly-emerging immune sequencing techniques, such as TCR-seq and BCR-seq, have been designed to sequence T and B cell receptors through quantitative PCR (qPCR). However, TCR-seq and BCR-seq are costly and not always available for particular datasets. To overcome this difficulty, <em>T</em>cr <em>R</em>eceptor <em>U</em>tilities for <em>S</em>olid <em>T</em>issue (TRUST) has been developed to infer the immune repertoire from bulk RNA-seq reads. V(D)J recombination results in sequences that do not align to genome references. TRUST combines reads that align to the V, D or J regions of the reference genome with sequences that do not align to the host genome to infer de novo assembled CDR3 sequences.</p>
<p>TRUST provides an overview of the immune repertoire of tumors, including CDR3 sequence length and the frequency of various V genes, J genes, and VJ pairs for different chains in the TCR and BCR. For cohort analysis, some basic metrics for TCRs and BCRs across samples could be computed to relate immune repertoire characteristics to specific phenotypes, including the fraction of reads mapped to TCR/BCR, the number of TCR/BCR unique clonotypes of CDR3 sequences, TCR/BCR diversity, and clonality. Since activated B cells undergo somatic hyper-mutation, antibody production, and Immunoglobulin (Ig) class switches during B cell maturation, TRUST also calculates the somatic hypermutation (SHM) rate for BCRs. In addition, TRUST predicts Ig isotypes for BCRs, which allows for quantification of the Ig compositions representing different maturation statuses (IGHM, IGHD, IGHG3, IGHG1, IGHA1, IGHG2, IGHG4, IGHA2, IGHE) and generation of an Ig class switches network. Taken together, the immune repertoire analysis quantifies the T/B cell population at both the individual and cohort levels, enabling one to characterize dynamic evolution in the TME via the diversity and clonal expansion of tumor-infiltrating T cells or B cells.</p>
<p>RIMA currently uses the fourth version of TRUST, TRUST4.</p>
<div id="trust4" class="section level2">
<h2><span class="header-section-number">5.1</span> TRUST4</h2>
<p>Example Command:</p>
<pre><code>./run-trust4 -b example.bam -f hg38_bcrtcr.fa --ref human_IMGT+C.fa</code></pre>
<p>The above command utilizes hg3_bcrtcr.fa (a fasta file containing the coordinates and sequences of V/D/J/C genes) and human_IMGT+C.fa (a V/D/J/C gene reference file from the IMGT database), and produces a simple report (example_report.tsv). This report contains a list of assembled clonotypes from example.bam with the following fields: “read_count, frequency(proportion of read_count), CDR3_dna, CDR3_amino_acids, V, D, J, C, consensus_id, consensus_id_full_length”. These fields contain the abundance, CDR3 sequence (nucleotides and amino acids), gene segment usage and identifier for each clonotype.</p>
<div id="trust4-output" class="section level3">
<h3><span class="header-section-number">5.1.1</span> TRUST4 output</h3>
<p>In the example codes below, we use one sample to illustrate how RIMA processes the results from TRUST4. RIMA directly adds the sample ID to the TRUST4 output file. This step makes it easier to combine all of individual sample results in the subsequent analysis. We provided the customized <a href="https://github.com/liulab-dfci/RIMA_pipeline/blob/master/src/immune_repertoire/trust4_metric_functions.R">trust4_metric_functions.R</a> in our pipeline repo to illustrate how BCR clusters, TCR and BCR clonality, BCR somatic hypermutation rate and BCR isotype class switching are calculated. Example individual outputs for sample ‘SRR8281233’ are included in each section below.</p>
<p><strong>Example output folder structure</strong>
<img src="images/Output_trust4.png" width="50%" style="display: block; margin: auto auto auto 0;" /></p>
</div>
</div>
<div id="cluster-calculation" class="section level2">
<h2><span class="header-section-number">5.2</span> Cluster calculation</h2>
<p>Activation of B cells leads to somatic hypermutation (<strong>SHM</strong>). As B cells proliferate, random mutations are generated at a high rate in the V-region sequence. Favorable mutations lead to better antigen binding and are selected for survival as the B cells continue to proliferate. BCR clustering is used to identify BCRs from the same lineage. The clusters can then be used to estimate somatic hypermutation rate (described later in this chapter).</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="Repertoire.html#cb35-1"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb35-2"><a href="Repertoire.html#cb35-2"></a></span>
<span id="cb35-3"><a href="Repertoire.html#cb35-3"></a><span class="co">#load processed function </span></span>
<span id="cb35-4"><a href="Repertoire.html#cb35-4"></a><span class="kw">source</span>(<span class="st">&quot;TRUST4/trust4_metric_functions.R&quot;</span>)</span>
<span id="cb35-5"><a href="Repertoire.html#cb35-5"></a></span>
<span id="cb35-6"><a href="Repertoire.html#cb35-6"></a><span class="co"># read the individual sample result from RIMA</span></span>
<span id="cb35-7"><a href="Repertoire.html#cb35-7"></a>cdr3 &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file =</span> <span class="st">&quot;TRUST4/SRR8281233_cdr3.out.processed.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb35-8"><a href="Repertoire.html#cb35-8"></a></span>
<span id="cb35-9"><a href="Repertoire.html#cb35-9"></a><span class="co">#filter out count of cdr ==0 and add the phenotype info for downstream comparison</span></span>
<span id="cb35-10"><a href="Repertoire.html#cb35-10"></a><span class="co">#This SRR8281233 smaple is Non-responder</span></span>
<span id="cb35-11"><a href="Repertoire.html#cb35-11"></a>phenotype &lt;-<span class="st"> &quot;NR&quot;</span></span>
<span id="cb35-12"><a href="Repertoire.html#cb35-12"></a>cdr3 &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, count <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb35-13"><a href="Repertoire.html#cb35-13"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">V =</span> <span class="kw">as.character</span>(V), <span class="dt">J =</span> <span class="kw">as.character</span>(J), <span class="dt">C =</span> <span class="kw">as.character</span>(C), <span class="dt">CDR3aa =</span> <span class="kw">as.character</span>(CDR3aa)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb35-14"><a href="Repertoire.html#cb35-14"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">clinic =</span> <span class="kw">as.character</span>(phenotype))</span>
<span id="cb35-15"><a href="Repertoire.html#cb35-15"></a></span>
<span id="cb35-16"><a href="Repertoire.html#cb35-16"></a><span class="kw">head</span>(cdr3)</span></code></pre></div>
<pre><code>##   count  frequency                                           CDR3nt
## 1   125 0.03682734                TGCATGCAAGCTCTACAAACTCCGTACACTTTT
## 2   124 0.03654850                TGTCAGCAGCGTAGCAACTGGCCTTGGACGTTC
## 3    98 0.05410145 TGTGCGATGGGGGATAGTAGTGGCTGGTACCGTCCTCCCGACTCCTGG
## 4    97 0.02847096                   TGTCAACATTACGGTACCTCGTGGACGTTC
## 5    79 0.02320237                TGTCAGCAGCGTAGCAACTGGCCGCTCACTTTC
## 6    78 0.02289418                TGCATGCAGCGTGTGAGTCTTCCCCACACTTTT
##             CDR3aa           V           D        J     C          cid
## 1      CMQALQTPYTF IGKV2-28*01           . IGKJ2*01  IGKC    assemble0
## 2      CQQRSNWPWTF IGKV3-11*01           . IGKJ1*01  IGKC assemble1701
## 3 CAMGDSSGWYRPPDSW IGHV3-33*01 IGHD6-19*01 IGHJ5*01 IGHG1    assemble1
## 4       CQHYGTSWTF IGKV3-20*01           . IGKJ1*01  IGKC assemble1713
## 5      CQQRSNWPLTF IGKV3-11*01           . IGKJ4*01  IGKC assemble1717
## 6      CMQRVSLPHTF IGKV2-40*01           . IGKJ2*01  IGKC assemble1719
##       sample clinic
## 1 SRR8281233     NR
## 2 SRR8281233     NR
## 3 SRR8281233     NR
## 4 SRR8281233     NR
## 5 SRR8281233     NR
## 6 SRR8281233     NR</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="Repertoire.html#cb37-1"></a><span class="co">#determine whether the cdr animo acid is complete or partial</span></span>
<span id="cb37-2"><a href="Repertoire.html#cb37-2"></a>cdr3<span class="op">$</span>is_complete &lt;-<span class="st"> </span><span class="kw">sapply</span>(cdr3<span class="op">$</span>CDR3aa, <span class="cf">function</span>(x) <span class="kw">ifelse</span>(x <span class="op">!=</span><span class="st"> &quot;partial&quot;</span> <span class="op">&amp;&amp;</span><span class="st"> </span>x <span class="op">!=</span><span class="st"> &quot;out_of_frame&quot;</span> <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;^_&quot;</span>,x) <span class="op">&amp;&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;^</span><span class="ch">\\</span><span class="st">?&quot;</span>, x),<span class="st">&quot;Y&quot;</span>,<span class="st">&quot;N&quot;</span>))</span>
<span id="cb37-3"><a href="Repertoire.html#cb37-3"></a></span>
<span id="cb37-4"><a href="Repertoire.html#cb37-4"></a><span class="co">#exact the TCR and BCR</span></span>
<span id="cb37-5"><a href="Repertoire.html#cb37-5"></a>cdr3.bcr &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, <span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG&quot;</span>,C))</span>
<span id="cb37-6"><a href="Repertoire.html#cb37-6"></a>cdr3.tcr &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3, <span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^TR&quot;</span>,C))</span>
<span id="cb37-7"><a href="Repertoire.html#cb37-7"></a></span>
<span id="cb37-8"><a href="Repertoire.html#cb37-8"></a><span class="co">#add lib size and clinic traits</span></span>
<span id="cb37-9"><a href="Repertoire.html#cb37-9"></a>cdr3.bcr &lt;-<span class="st"> </span>cdr3.bcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">lib.size =</span> <span class="kw">sum</span>(count)) </span>
<span id="cb37-10"><a href="Repertoire.html#cb37-10"></a>cdr3.tcr &lt;-<span class="st"> </span>cdr3.tcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">lib.size =</span> <span class="kw">sum</span>(count)) </span>
<span id="cb37-11"><a href="Repertoire.html#cb37-11"></a></span>
<span id="cb37-12"><a href="Repertoire.html#cb37-12"></a><span class="co">#split BCR into heavy chain and light chain</span></span>
<span id="cb37-13"><a href="Repertoire.html#cb37-13"></a>cdr3.bcr.heavy &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3.bcr, <span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IGH&quot;</span>,C))</span>
<span id="cb37-14"><a href="Repertoire.html#cb37-14"></a>cdr3.bcr.light &lt;-<span class="st"> </span><span class="kw">subset</span>(cdr3.bcr, <span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,V) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,J) <span class="op">|</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^IG[K|L]&quot;</span>,C))</span>
<span id="cb37-15"><a href="Repertoire.html#cb37-15"></a></span>
<span id="cb37-16"><a href="Repertoire.html#cb37-16"></a><span class="co">#save BCR and TCR info for downsteam use</span></span>
<span id="cb37-17"><a href="Repertoire.html#cb37-17"></a>outdir &lt;-<span class="st"> &quot;TRUST4/individual/&quot;</span></span>
<span id="cb37-18"><a href="Repertoire.html#cb37-18"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb37-19"><a href="Repertoire.html#cb37-19"></a><span class="kw">save</span>(cdr3.bcr.light, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_BCR_light.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb37-20"><a href="Repertoire.html#cb37-20"></a><span class="kw">save</span>(cdr3.bcr.heavy, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_BCR_heavy.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb37-21"><a href="Repertoire.html#cb37-21"></a><span class="kw">save</span>(cdr3.tcr, <span class="dt">file =</span> <span class="kw">paste</span>(outdir, <span class="st">&quot;TRUST4_TCR.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb37-22"><a href="Repertoire.html#cb37-22"></a></span>
<span id="cb37-23"><a href="Repertoire.html#cb37-23"></a><span class="co">#BCR clustering </span></span>
<span id="cb37-24"><a href="Repertoire.html#cb37-24"></a><span class="co">#Note that not every sample have BCR cluster</span></span>
<span id="cb37-25"><a href="Repertoire.html#cb37-25"></a>sample_bcr_cluster &lt;-<span class="st"> </span><span class="kw">BuildBCRlineage</span>(<span class="dt">sampleID =</span> sample, <span class="dt">Bdata =</span> cdr3.bcr.heavy, <span class="dt">start=</span><span class="dv">3</span>, <span class="dt">end=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix
## use default substitution matrix
## use default substitution matrix</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="Repertoire.html#cb39-1"></a><span class="kw">save</span>(sample_bcr_cluster,<span class="dt">file =</span> <span class="kw">paste</span>(outdir, sample,<span class="st">&quot;_TRUST4_BCR_heavy_cluster.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb39-2"><a href="Repertoire.html#cb39-2"></a></span>
<span id="cb39-3"><a href="Repertoire.html#cb39-3"></a><span class="kw">attributes</span>(sample_bcr_cluster)</span></code></pre></div>
<pre><code>## $names
## [1] &quot;RGSPRFID&quot; &quot;KVPGRRTR&quot; &quot;RNLYYGAY&quot;</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="Repertoire.html#cb41-1"></a><span class="kw">head</span>(sample_bcr_cluster<span class="op">$</span>RGSPRFID)</span></code></pre></div>
<pre><code>## $distMat
##      [,1] [,2] [,3] [,4]
## [1,]    0    1    1    3
## [2,]    0    0    2    4
## [3,]    0    0    0    2
## [4,]    0    0    0    0
## 
## $Sequences
## [1] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG&quot;
## [2] &quot;TGTGCGCGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG&quot;
## [3] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCTGG&quot;
## [4] &quot;TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCCAG&quot;
## 
## $data
##   count   frequency                                                 CDR3nt
## 1     2 0.001099511 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG
## 2     2 0.001099511 TGTGCGCGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCCTTCACTTCTGG
## 3     7 0.003848289 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCTGG
## 4     2 0.001099511 TGTGCGAGAGGATCCCCCCGGTTCATTGACTACGGTGGTTCCTTTCACTTCCAG
##               CDR3aa           V           D        J     C         cid
## 1 CARGSPRFIDYGGSLHFW IGHV4-31*01 IGHD4-23*01 IGHJ4*02 IGHG1 assemble836
## 2 CARGSPRFIDYGGSLHFW IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
## 3 CARGSPRFIDYGGSFHFW IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
## 4 CARGSPRFIDYGGSFHFQ IGHV4-31*11 IGHD4-23*01 IGHJ4*02 IGHG1  assemble90
##       sample clinic is_complete lib.size
## 1 SRR8281233     NR           Y     5184
## 2 SRR8281233     NR           Y     5184
## 3 SRR8281233     NR           Y     5184
## 4 SRR8281233     NR           Y     5184</code></pre>
</div>
<div id="clonotypes-per-kilo-reads-cpk" class="section level2">
<h2><span class="header-section-number">5.3</span> Clonotypes per kilo-reads (CPK)</h2>
<p>CPK can be calculated by taking the number of unique CDR3 calls for a chain divided by the total number of reads for that chain, with this result being multiplied by 1000. This is used as a measure of clonotype diversity.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="Repertoire.html#cb43-1"></a><span class="co">#TCR CPK</span></span>
<span id="cb43-2"><a href="Repertoire.html#cb43-2"></a>cpk &lt;-<span class="st"> </span><span class="kw">aggregate</span>(CDR3aa <span class="op">~</span><span class="st"> </span>sample<span class="op">+</span>clinic<span class="op">+</span>lib.size, cdr3.tcr, <span class="cf">function</span>(x) <span class="kw">length</span>(<span class="kw">unique</span>(x))) <span class="op">%&gt;%</span></span>
<span id="cb43-3"><a href="Repertoire.html#cb43-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CPK =</span> <span class="kw">signif</span>(CDR3aa<span class="op">/</span>(lib.size<span class="op">/</span><span class="dv">1000</span>),<span class="dv">4</span>))</span>
<span id="cb43-4"><a href="Repertoire.html#cb43-4"></a>cpk</span></code></pre></div>
<pre><code>##       sample clinic lib.size CDR3aa CPK
## 1 SRR8281233     NR       73     20 274</code></pre>
</div>
<div id="tcr-and-bcr-entropy-and-clonality" class="section level2">
<h2><span class="header-section-number">5.4</span> TCR and BCR entropy and Clonality</h2>
<p>The Shannon entropy index is a measure used for repertoire diversity using clonotype frequencies, which reflects both richness and evenness of the repertoire. This measure informs us of the probability that two random selections from the same repertoire would represent the same clonotype.</p>
<p>Clonality can be measured using normalized entropy over the number of unique clones (1-shannon entropy/log(N), where N is the number of unique clones). It is equivalent to 1 - Pielou’s Evenness, making it inversely proportional to diversity. A higher clonality index indicates an uneven repertoire due to expansion of clones.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="Repertoire.html#cb45-1"></a><span class="co">#BCR clonality and entropy</span></span>
<span id="cb45-2"><a href="Repertoire.html#cb45-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb45-3"><a href="Repertoire.html#cb45-3"></a></span>
<span id="cb45-4"><a href="Repertoire.html#cb45-4"></a>single_sample_bcr_clonality &lt;-<span class="st"> </span><span class="kw">getClonality</span>(sample, cdr3.bcr.heavy, <span class="dt">start=</span><span class="dv">3</span>, <span class="dt">end=</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## use default substitution matrix
## use default substitution matrix
## use default substitution matrix
## [1] &quot;SRR8281233&quot;</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="Repertoire.html#cb47-1"></a>single_sample_bcr_clonality[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##               sample            clonality              entropy 
##         &quot;SRR8281233&quot; &quot;0.0899083128267854&quot;   &quot;6.83997594611138&quot;</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="Repertoire.html#cb49-1"></a><span class="co">#TCR clonality and entropy</span></span>
<span id="cb49-2"><a href="Repertoire.html#cb49-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb49-3"><a href="Repertoire.html#cb49-3"></a></span>
<span id="cb49-4"><a href="Repertoire.html#cb49-4"></a>single_sample_tcr_clonality &lt;-<span class="st"> </span><span class="kw">getClonalityTCR</span>(sample,cdr3.tcr)</span>
<span id="cb49-5"><a href="Repertoire.html#cb49-5"></a>single_sample_tcr_clonality</span></code></pre></div>
<pre><code>##               sample            clonality              entropy 
##         &quot;SRR8281233&quot; &quot;0.0699534783052342&quot;    &quot;4.0850595412347&quot;</code></pre>
</div>
<div id="fraction-of-reads-mapped-to-tcr-and-bcr" class="section level2">
<h2><span class="header-section-number">5.5</span> Fraction of reads mapped to TCR and BCR</h2>
<p>The fraction of reads mapped to TCR and BCR can be used as a rough approximation of T and B cell infiltration. The fraction of all reads that are mapped to either TCR or BCR is indicated by the V,D,J-gene prefixes of the clonotypes identified by TRUST4. The fraction is composed of the proportion of reads with genes prefixed by “TR” (TCR) and genes prefixed by “IG” (BCR) divided by the total of all mapped genes in the sample.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="Repertoire.html#cb51-1"></a><span class="co">#exact the mapped reads info from alignment index file </span></span>
<span id="cb51-2"><a href="Repertoire.html#cb51-2"></a>sample &lt;-<span class="st"> &quot;SRR8281233&quot;</span></span>
<span id="cb51-3"><a href="Repertoire.html#cb51-3"></a></span>
<span id="cb51-4"><a href="Repertoire.html#cb51-4"></a>file &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;TRUST4/SRR8281233.sorted.bam.stat.txt&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb51-5"><a href="Repertoire.html#cb51-5"></a>mapped.reads &lt;-<span class="st"> </span>file[<span class="st">&quot;reads mapped:&quot;</span>,<span class="st">&quot;V2&quot;</span>]</span>
<span id="cb51-6"><a href="Repertoire.html#cb51-6"></a>individual.stats &lt;-<span class="st"> </span><span class="kw">cbind.data.frame</span>(<span class="dt">sample =</span> sample, <span class="dt">map.reads =</span> mapped.reads)</span>
<span id="cb51-7"><a href="Repertoire.html#cb51-7"></a></span>
<span id="cb51-8"><a href="Repertoire.html#cb51-8"></a><span class="co">#---------fraction of BCR reads------------------</span></span>
<span id="cb51-9"><a href="Repertoire.html#cb51-9"></a><span class="co">##extract library size</span></span>
<span id="cb51-10"><a href="Repertoire.html#cb51-10"></a>lib.size &lt;-<span class="st"> </span>cdr3.bcr.heavy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span></span>
<span id="cb51-11"><a href="Repertoire.html#cb51-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">lib =</span> <span class="kw">mean</span>(lib.size))</span>
<span id="cb51-12"><a href="Repertoire.html#cb51-12"></a></span>
<span id="cb51-13"><a href="Repertoire.html#cb51-13"></a><span class="co">##combine stats and library size</span></span>
<span id="cb51-14"><a href="Repertoire.html#cb51-14"></a>bcr.lib.reads &lt;-<span class="st"> </span><span class="kw">merge</span>(individual.stats,lib.size,<span class="dt">by =</span> <span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-15"><a href="Repertoire.html#cb51-15"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Infil =</span> <span class="kw">signif</span>(<span class="kw">as.numeric</span>(lib)<span class="op">/</span><span class="kw">as.numeric</span>(map.reads),<span class="dv">4</span>))</span>
<span id="cb51-16"><a href="Repertoire.html#cb51-16"></a></span>
<span id="cb51-17"><a href="Repertoire.html#cb51-17"></a><span class="co">#------------fraction of TCR reads-----------------</span></span>
<span id="cb51-18"><a href="Repertoire.html#cb51-18"></a><span class="co">##extract library size</span></span>
<span id="cb51-19"><a href="Repertoire.html#cb51-19"></a>lib.size &lt;-<span class="st"> </span>cdr3.tcr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(sample) <span class="op">%&gt;%</span></span>
<span id="cb51-20"><a href="Repertoire.html#cb51-20"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">lib =</span> <span class="kw">mean</span>(lib.size)) </span>
<span id="cb51-21"><a href="Repertoire.html#cb51-21"></a><span class="co">##combine stats and library size</span></span>
<span id="cb51-22"><a href="Repertoire.html#cb51-22"></a>tcr.lib.reads &lt;-<span class="st"> </span><span class="kw">merge</span>(individual.stats,lib.size,<span class="dt">by =</span> <span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-23"><a href="Repertoire.html#cb51-23"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Infil =</span> <span class="kw">signif</span>(<span class="kw">as.numeric</span>(lib)<span class="op">/</span><span class="kw">as.numeric</span>(map.reads),<span class="dv">4</span>))</span>
<span id="cb51-24"><a href="Repertoire.html#cb51-24"></a></span>
<span id="cb51-25"><a href="Repertoire.html#cb51-25"></a>combined &lt;-<span class="st"> </span><span class="kw">rbind</span>(bcr.lib.reads, tcr.lib.reads) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Type =</span> <span class="kw">c</span>(<span class="st">&quot;BCR&quot;</span>, <span class="st">&quot;TCR&quot;</span>))</span>
<span id="cb51-26"><a href="Repertoire.html#cb51-26"></a>combined</span></code></pre></div>
<pre><code>##       sample map.reads  lib     Infil Type
## 1 SRR8281233  59972094 5184 8.644e-05  BCR
## 2 SRR8281233  59972094   73 1.217e-06  TCR</code></pre>
</div>
<div id="somatic-hypermutation-rate-of-bcr" class="section level2">
<h2><span class="header-section-number">5.6</span> Somatic hypermutation rate of BCR</h2>
<p>As described in the BCR clustering section of this chapter, somatic hypermutation (<strong>SHM</strong>) introduces more variation in a particular BCR and occurs when mature B cells are activated and proliferate. For each BCR cluster identified by TRUST4, TRUST4 calculates the number of point mutations in the nucleotide sequences in order to estimate SHM. SHM can be used as an approximation of B cell activity within a sample.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="Repertoire.html#cb53-1"></a>SHM.ratio &lt;-<span class="st"> </span><span class="kw">getSHMratio</span>(sample_bcr_cluster)</span></code></pre></div>
<pre><code>## [1] 54
## [1] 2
## [1] 48
## [1] 3
## [1] 51
## [1] 1</code></pre>
</div>
<div id="ig-isotype-frequency" class="section level2">
<h2><span class="header-section-number">5.7</span> Ig isotype frequency</h2>
<p>The immunoglobulin heavy (IGH) locus includes the constant heavy (IGHC) gene segment. This gene segment includes different isotypes (IGHA1, IGHA2, IGHD, IGHE, IGHG1, IGHG2, IGHG3, IGHG4, IGHM) and the proportion of the abundance of these different segments indicates the isotype frequency. The isotype frequency can be found by isolating reads with genes prefixed “IGH” and comparing the occurrence of different isotypes in the C_gene column.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="Repertoire.html#cb55-1"></a><span class="co">#This SRR8281233 sample is Non-responder</span></span>
<span id="cb55-2"><a href="Repertoire.html#cb55-2"></a>phenotype &lt;-<span class="st"> &quot;NR&quot;</span></span>
<span id="cb55-3"><a href="Repertoire.html#cb55-3"></a></span>
<span id="cb55-4"><a href="Repertoire.html#cb55-4"></a>st.Ig &lt;-<span class="st"> </span>cdr3.bcr.heavy <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-5"><a href="Repertoire.html#cb55-5"></a><span class="st">  </span><span class="kw">group_by</span>(clinic,sample) <span class="op">%&gt;%</span></span>
<span id="cb55-6"><a href="Repertoire.html#cb55-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">est_clonal_exp_norm =</span> frequency<span class="op">/</span><span class="kw">sum</span>(frequency)) <span class="op">%&gt;%</span><span class="st">   </span><span class="co">#as.numeric(sample.clones[filename,2])</span></span>
<span id="cb55-7"><a href="Repertoire.html#cb55-7"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(C <span class="op">!=</span><span class="st"> &quot;*&quot;</span> <span class="op">&amp;</span><span class="st"> </span>C <span class="op">!=</span><span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb55-8"><a href="Repertoire.html#cb55-8"></a><span class="st">  </span><span class="kw">group_by</span>(sample, C) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-9"><a href="Repertoire.html#cb55-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(<span class="dt">Num.Ig =</span> <span class="kw">sum</span>(est_clonal_exp_norm)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">clinic =</span> phenotype)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;sample&#39;. You can override using the `.groups` argument.</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="Repertoire.html#cb57-1"></a>st.Ig</span></code></pre></div>
<pre><code>## # A tibble: 8 x 4
## # Groups:   sample [1]
##   sample     C      Num.Ig clinic
##   &lt;chr&gt;      &lt;chr&gt;   &lt;dbl&gt; &lt;chr&gt; 
## 1 SRR8281233 IGHA1 0.134   NR    
## 2 SRR8281233 IGHA2 0.0682  NR    
## 3 SRR8281233 IGHD  0.00385 NR    
## 4 SRR8281233 IGHG1 0.542   NR    
## 5 SRR8281233 IGHG2 0.0704  NR    
## 6 SRR8281233 IGHG3 0.0280  NR    
## 7 SRR8281233 IGHG4 0.0126  NR    
## 8 SRR8281233 IGHM  0.0407  NR</code></pre>
</div>
<div id="cohort-analysis" class="section level2">
<h2><span class="header-section-number">5.8</span> Cohort analysis</h2>
<p>The examples above demonstrate the basic immune repertoire-related metrics using a single sample. In the cohort analysis, all of samples are combined in order to facilitate comparisons within a particular phenotype. The phenotype used for comparison is the one identified in the config.yaml cohort parameter section. This section is described in more detail in Chapter 4.2.1.
.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="Repertoire.html#cb59-1"></a><span class="co">#load the pre-processed results that contain all samples </span></span>
<span id="cb59-2"><a href="Repertoire.html#cb59-2"></a><span class="co">#These results would be automatically generated after running the RIMA immune repertoire module</span></span>
<span id="cb59-3"><a href="Repertoire.html#cb59-3"></a></span>
<span id="cb59-4"><a href="Repertoire.html#cb59-4"></a>inputdir &lt;-<span class="st"> &quot;TRUST4/Cohort/&quot;</span></span>
<span id="cb59-5"><a href="Repertoire.html#cb59-5"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_cluster.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-6"><a href="Repertoire.html#cb59-6"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_clonality.Rdata&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-7"><a href="Repertoire.html#cb59-7"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_SHMRatio.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-8"><a href="Repertoire.html#cb59-8"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-9"><a href="Repertoire.html#cb59-9"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_Ig_CS.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-10"><a href="Repertoire.html#cb59-10"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-11"><a href="Repertoire.html#cb59-11"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_BCR_heavy.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-12"><a href="Repertoire.html#cb59-12"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR_lib_reads_Infil.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-13"><a href="Repertoire.html#cb59-13"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-14"><a href="Repertoire.html#cb59-14"></a><span class="kw">load</span>(<span class="kw">paste</span>(inputdir,<span class="st">&quot;TRUST4_TCR_clonality.Rdata&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb59-15"><a href="Repertoire.html#cb59-15"></a></span>
<span id="cb59-16"><a href="Repertoire.html#cb59-16"></a></span>
<span id="cb59-17"><a href="Repertoire.html#cb59-17"></a><span class="co">#call the ploting function</span></span>
<span id="cb59-18"><a href="Repertoire.html#cb59-18"></a><span class="kw">source</span>(<span class="st">&quot;TRUST4/trust4_plot.R&quot;</span>)</span>
<span id="cb59-19"><a href="Repertoire.html#cb59-19"></a>meta &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;metasheet.csv&quot;</span>)</span>
<span id="cb59-20"><a href="Repertoire.html#cb59-20"></a>p &lt;-<span class="st"> </span><span class="kw">Trust4_plot</span>(<span class="dt">phenotype =</span> <span class="st">&quot;Responder&quot;</span>, <span class="dt">metasheet =</span> meta)</span>
<span id="cb59-21"><a href="Repertoire.html#cb59-21"></a></span>
<span id="cb59-22"><a href="Repertoire.html#cb59-22"></a>p</span></code></pre></div>
<p><img src="RIMA-tutorial_files/figure-html/unnamed-chunk-27-1.png" width="576" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Differential.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Infiltration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RIMA-tutorial.pdf", "RIMA-tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
